---
title: "PRAC Paired Design"
author: "PRAC Team: A. Auladell, J. Martí & D. Mas"
date: "1 de junio de 2016"
output:
  html_document:
    mathjax: local
    self_contained: false
bibliography: prac.bib
---


```{r setup, cache=FALSE, echo=FALSE, message=FALSE, warning=FALSE, tidy=FALSE}
library(knitr)
opts_chunk$set(cache=T, fig.align='center', dpi=100, message=FALSE)

```

The following are the main packages used along the methods path. The data 
is readed from [here]() and stored in a `SummarizedExperiment`container from 
@SE .

```{r loading packages and reading data, echo=TRUE, warning= FALSE, error=FALSE, message = TRUE, tidy=FALSE}
library("SummarizedExperiment")
library("edgeR")
library("geneplotter")
library("ggplot2")
library("sva")
library("pheatmap")
library("RColorBrewer")
library("Rgraphviz")

old.par <- par(mar = c(0, 0, 0, 0)) # to get CLI 

prac.se <- readRDS("data/sePRAD.rds")

```

# Filtering and Data description

The original data set obtained from @Rahman15112015 have the following
distribution of samples. 

```{r original dataset}

table(prac.se$type)

```

The design of our experiment is based on paired samples. Therefore, each 
patient have to be the source for an assigned as tumor sample and a assigned
normal sample. 

```{r filtering for the paired data}
# Subset the paired samples
## we build a table from the bcr barcode occurences in our dataset
n_occur <- data.frame(table(colData(prac.se)$bcr_patient_barcode))

## we filter out the na patients (having a na in the barcode)
## and the ones that occur only 1 time (they are unique)
prac.se.dupl <- prac.se[,colData(prac.se)$bcr_patient_barcode
                 %in% n_occur$Var1[n_occur$Freq > 1] & 
                 !is.na(colData(prac.se)$bcr_patient_barcode)]
## See the results.
table(colData(prac.se.dupl)$type)

```

The selected samples are 50 normal and 50 tumor. Most of the Normal samples had a paired tumor sample but 2. We proceed in the normalization of the data using the `DGE` package @Robinson2010 .

```{r Normalizing}

prac.dge <- DGEList(counts = assays(prac.se.dupl)$counts, 
                    genes = mcols(prac.se.dupl), group = prac.se.dupl$type)
prac.dgenorm <- calcNormFactors(prac.dge)
assays(prac.se.dupl)$logCPM <- cpm(prac.dgenorm, log=TRUE, prior.count=3.5)
assays(prac.se.dupl)$logCPM[1:5, 1:5]
```

```{r Multidensity, eval = FALSE, fig.align='center', fig.cap='Fig 1: Multidensity plot representing the clusters of the non expressed genes at the left and the expressed genes at the right.', echo=FALSE, fig.width=12}
MDS_normal <- prac.dgenorm[,prac.dgenorm$samples$group == 'normal']
MDS_tumor <- prac.dgenorm[,prac.dgenorm$samples$group == 'tumor']

logCPM.MDS_normal <- cpm(MDS_normal, log = TRUE, prior.count = 3.25)
logCPM.MDS_tumor <- cpm(MDS_tumor, log = TRUE, prior.count = 3.25)

par(mfrow=c(1,2))
multidensity(as.list(as.data.frame(logCPM.MDS_normal)), xlab = "log2 CPM", legend = NULL, main = "Normal samples")
multidensity(as.list(as.data.frame(logCPM.MDS_tumor)), xlab = "log2 CPM", legend = NULL, main = "Tumor samples")

```


By looking at the disribution we observe a peak around logCPM 1. ?¿?¿?¿?¿?

To avoid errors or misleading data in the future we set up a threshold of the gene expression to remove the low expressed genes. For these values the CPM expression level is very low and we cannot establish real conclusions.


```{r filter genes}
mask <- rowMeans(assays(prac.se.dupl)$logCPM) > 1
prac.se.dupl <- prac.se.dupl[mask, ]
prac.dgenorm <- prac.dgenorm[mask, ]
```

After filtering the low-expressed genes we explore the data to perform a Quality Assesment.
We explore the data to check whether there exists some kind of batch effect or an undesired source of variation. See fig 2,3 and 4 below.

** Ojo, hi ha chunks sense processar, activar al final**

Here we asses the Library size for each sample. Although we can see an extreme value in the rigth end, the distribution is uniform.

```{r lib size plots, eval = FALSE, fig.cap='Fig 2: Library size distribution of the samples.'}

# Libsize distribution of the samples
libsize <- data.frame(libsize = prac.dgenorm$sample$lib.size/1e6, type = prac.se.dupl$type, samples = rownames(prac.dgenorm$samples) )
ggplot(libsize, aes(x=reorder(samples, libsize), y=libsize, fill=type)) +  theme(plot.background = element_blank(), axis.title.x=element_blank(), axis.text.x = element_blank()) + geom_bar(stat = "identity") 

```



<center><h4>Tumor samples MA-plot</h4></center>

```{r maPlotsTumor, eval= FALSE, fig.align="center", fig.height= 20, fig.width=12, dpi=100, echo=FALSE, fig.cap="Fig. 3: MA-plots of the tumor samples."}

setmp <- prac.se.dupl[, prac.se.dupl$type == "tumor"]
dgetmp <- prac.dgenorm[, prac.dgenorm$type == "tumor"]

samples.number <- length(rownames(colData(setmp)))
if (samples.number %% 6 == 0) { 
  nr <- samples.number / 6
} else {nr <- samples.number %/% 6 + 1}
par(mfrow=c(nr, 6), mar=c(4, 5, 3, 1))

for (i in 1:ncol(setmp)) {
  A <- rowMeans(assays(setmp)$logCPM)
  M <- assays(setmp)$logCPM[, i] - A
  samplename <- substr(as.character(setmp$bcr_patient_barcode[i]), 1, 12)
  smoothScatter(A, M, main=samplename, las=1)
  abline(h=0, col="blue", lwd=2)
  lo <- lowess(M ~ A)
  lines(lo$x, lo$y, col="red", lwd=2)
}
```

<center><h4>Normal samples MA-plot</h4></center>

```{r maPlotsNormal, eval = FALSE, fig.align="center", fig.height= 20, fig.width=12, dpi=100, echo=FALSE, fig.cap="Fig. 4: MA-plots of the tumor samples."}

setmp <- prac.se.dupl[, prac.se.dupl$type == "normal"]
dgetmp <- prac.dgenorm[, prac.dgenorm$type == "normal"]

samples.number <- length(rownames(colData(setmp)))
if (samples.number %% 6 == 0) { 
  nr <- samples.number / 6
} else {nr <- samples.number %/% 6 + 1}
par(mfrow=c(nr, 6), mar=c(4, 5, 3, 1))

for (i in 1:ncol(setmp)) {
  A <- rowMeans(assays(setmp)$logCPM)
  M <- assays(setmp)$logCPM[, i] - A
  samplename <- substr(as.character(setmp$bcr_patient_barcode[i]), 1, 12)
  smoothScatter(A, M, main=samplename, las=1)
  abline(h=0, col="blue", lwd=2)
  lo <- lowess(M ~ A)
  lines(lo$x, lo$y, col="red", lwd=2)
}
```

From the plots above, we see that in figure 1 the distribution of library size presents some stratifiction, with values of 20 M and a mean of 40 M. This changes in lib size will be compensated by the normalization. In figure 2, we observe that the distribution of expression is the expressed, identical in both graphs. Finally, in figure 3 and 4 we present the MA plots, the tendency line representing the mean of normalized values (red line) is reasonably close to the expected tendency line (blue line), which is centered at mean 0, thus normalization procedure has been succesfully applied.

# Batch effect analysis

From all the potential soucre of batch effect in an experiment of this magnitude, one of the most ocasional is TSS. We proceed to analyse the sample distribution in the different processing steps of the samples.

```{r batch effect, echo = FALSE}

tss <- substr(colnames(prac.se.dupl), 6, 7)
center <- substr(colnames(prac.se.dupl), 27, 28)
plate <- substr(colnames(prac.se.dupl), 22, 25)
portionanalyte <- substr(colnames(prac.se.dupl), 18, 20)
samplevial <- substr(colnames(prac.se.dupl), 14, 16)

table(data.frame(TYPE=prac.se.dupl$type, Center=center))
table(data.frame(TYPE=prac.se.dupl$type, TSS=tss))
table(data.frame(TYPE=prac.se.dupl$type, Sample_vial=samplevial))
table(data.frame(TYPE=prac.se.dupl$type, Portion_analyte=portionanalyte))
table(data.frame(TYPE=prac.se.dupl$type, Plate=plate))

```

By looking at the distributions of the tables, we observe that the values are equally distributed. It is expected to not present any batch effect. One of the possible batch effects would be the plate distribution, but in relation of the type variable they are equally analysed.

```{r dendogram, echo=FALSE, fig.width=10, fig.cap="Fig. 5: Dendogram of Spearman distances."}
par(old.par)
logCPM <- assays(prac.se.dupl)$logCPM
d <- as.dist(1-cor(logCPM, method="spearman"))
sampleClustering <- hclust(d)
batch <- as.integer(factor(tss))
colors <- palette()
palette(c(colors,"darkorange1","darkmagenta","darkolivegreen","red3","seagreen","pink1","yellow3","sienna1"))
sampleDendrogram <- as.dendrogram(sampleClustering, hang=0.1)
names(batch) <- colnames(prac.se.dupl)
outcome <- paste(substr(colnames(prac.se.dupl), 9, 12), as.character(prac.se.dupl$type), sep="-")
names(outcome) <- colnames(prac.se.dupl)
sampleDendrogram <- dendrapply(sampleDendrogram,
                               function(x, batch, labels) {
                                 if (is.leaf(x)) {
                                   attr(x, "nodePar") <- list(lab.col=as.vector(batch[attr(x, "label")]))
                                   attr(x, "label") <- as.vector(labels[attr(x, "label")])
                                 }
                                 x
                               }, batch, outcome)

plot(sampleDendrogram, main="Hierarchical clustering of samples")
legend("topright", paste("Batch", sort(unique(batch)), levels(factor(tss))), fill=sort(unique(batch)))

```

```{r clustering, fig.height=10, fig.width=10, , fig.cap="Fig. 6: Heatmap of Spearman distances."}
sampleDists <- d
head(sampleDists)
class(sampleDists)


sampleDistMatrix <- as.matrix( sampleDists )
rownames(sampleDistMatrix) <- paste( substr(colnames(prac.se.dupl), 9, 12), prac.se.dupl$type, sep="-" )
colnames(sampleDistMatrix) <- NULL
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
pheatmap(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists,
         col=colors)
```

As a general insight from our data we can see how samples are clustering by type. No TSS cluster can be observed so we can be almost sure this factor is not introducing us some batch effect. In the general dendogram the samples are colored by TSS. Furthermore, in the heatmap we can see a clear clustering by type although some of the samples have an undesired variation with the rest of it. In order to asses that anormal cluster we will plot a MDS to detect those samples. 

The clustering have been done both in the Dendogram and the heatmap using the Spearman method. The heatmap clustering have been done using the @pheatmap package.

```{r mds, echo=FALSE, fig.align='center', fig.width=14, dpi=100, fig.cap='Fig. 9: MDS of the prac.se.dupl dataset', fig.height=8}
par(mfrow=c(1, 1), mar=c(0,0,0,0))
plotMDS(prac.dgenorm, labels=outcome, col=batch)
legend("right", paste("Batch", sort(unique(batch)), levels(factor(tss))),
       fill=sort(unique(batch)), inset=0.05)
```

```{r mds2, echo=FALSE, fig.align='center', fig.width=14, dpi=100, fig.cap='Fig. 10: MDS of the prac.se.dupl dataset', fig.height=8}
type <- as.integer(factor(prac.se.dupl$type))
plotMDS(prac.dgenorm, pch = 19, col=type)
legend("right", paste("Type", sort(unique(type)), levels(factor(prac.se.dupl$type))),
       fill=sort(unique(type)), inset=0.05)
```

We observe an anomalous cluster of normal samples. In order to minimize the associated variability of the samples and focus only in the variability responsible for the generation of the tumor we filter those samples.

```{r filter those ind}
GetID <- function(number, se){
  id <- grep(number,rownames(colData(se)))
  return(rownames(colData(se))[id])
}
number.id <- c("7737","7745","7738","7211","7747","6356","6362","5769") 
pruning.vec <- as.character(sapply(number.id, GetID, prac.se.dupl))
prac.dgenorm <- prac.dgenorm[,
                   !rownames(prac.dgenorm$samples) %in% pruning.vec]
table(prac.dgenorm$samples$group)
prac.se.dupl <- prac.se.dupl[,rownames(prac.dgenorm$samples)]
table(colData(prac.se.dupl)$type)
```

After pruning those samples out we can plot again the heatmap and the MDS to see weather the results have improved. 

```{r Custering 2, echo = FALSE, fig.height=10, fig.width=10, fig.cap="Fig. 7: Heatmap of Spearman distances without the disturbing samples."}

logCPM <- assays(prac.se.dupl)$logCPM
d <- as.dist(1-cor(logCPM, method="spearman"))

sampleDists <- d
head(sampleDists)
class(sampleDists)


sampleDistMatrix <- as.matrix( sampleDists )
rownames(sampleDistMatrix) <- paste( substr(colnames(prac.se.dupl), 9, 12), prac.se.dupl$type, sep="-" )
colnames(sampleDistMatrix) <- NULL
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
pheatmap(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists,
         col=colors)

```

```{r MDS after prunning, echo=FALSE}
tss <- substr(colnames(prac.se.dupl), 6, 7)
batch <- as.integer(factor(tss))
outcome <- paste(substr(colnames(prac.se.dupl), 9, 12), as.character(prac.se.dupl$type), sep="-")
names(outcome) <- colnames(prac.se.dupl)
```

```{r mds3, echo=FALSE, fig.align='center', fig.width=14, dpi=100, fig.cap='Fig. 8: MDS of the prac.se.dupl dataset', fig.height=8}
plotMDS(prac.dgenorm, labels=outcome, col=batch)
legend("topright", paste("Batch", sort(unique(batch)), levels(factor(tss))),
       fill=sort(unique(batch)), inset=0.05)


```

```{r mds4, echo=FALSE, fig.align='center', fig.width=14, dpi=100, fig.cap='Fig. 9: MDS of the prac.se.dupl dataset', fig.height=8}

type <- as.integer(factor(prac.se.dupl$type))
plotMDS(prac.dgenorm, pch = 19, col=type)
legend("topleft", paste("Type", sort(unique(type)), levels(factor(prac.se.dupl$type))),
       fill=sort(unique(type)), inset=0.05)
```

Having the samples / genes filtered we can finnish the Quality asesment of the data and proceed with the differential expression analysis. 

# Differential Expression Analysis

Initially, we observe the differental expression with a multiple linear regression test, having as a cutoff a FDR of 0.001 .

```{r DE1}
colData(prac.se.dupl)$bcr_patient_barcode <- droplevels(colData(prac.se.dupl)$bcr_patient_barcode)
design <- model.matrix(~ type , data=colData(prac.se.dupl))
head(design)
tail(design)
fit <- lmFit(assays(prac.se.dupl)$logCPM, design)
fit <- eBayes(fit)
FDRcutoff <- 0.001
res <- decideTests(fit, p.value=FDRcutoff)
summary(res)
genesmd <- data.frame(chr=as.character(seqnames(rowRanges(prac.se.dupl))),
                      symbol=as.character(rowRanges(prac.se.dupl)[, 1]),
                      stringsAsFactors=FALSE)
fit$genes <- genesmd
tt <- topTable(fit, coef=2, n=Inf)
head(tt)
sort(table(tt$chr[tt$adj.P.Val < FDRcutoff]), decreasing=TRUE)


par(mfrow=c(1,2), mar=c(4, 5, 2, 2))
hist(tt$P.Value, xlab="Raw P-values", main="", las=1)
qqt(fit$t[, 2], df=fit$df.prior+fit$df.residual, main="", pch=".", cex=3)
qqline(fit$t[, 2], col=2, lwd=3)
abline(0, 1, lwd=3)

```

In black we can see the Normal distribution ( \(\sigma = 1 \) ) and the red is the `qqline` extracted from the distribution itself.

Right now we have a real true candidate overexpressed in tumor at the right top. The other genes, although with a high variability seem to follow a normal distribution of \(\sigma = 4 \).

## Improving the DEA with voom and patient barcode

```{r DE voom}
par(mfrow=c(1,1))
design <- model.matrix(~ type + bcr_patient_barcode , data=colData(prac.se.dupl))
v <- voom(prac.dgenorm, design, plot=TRUE)

# redo the process with the voom weight

fit2 <- lmFit(v, design)
fit2 <- eBayes(fit2)
res2 <- decideTests(fit2, p.value=FDRcutoff)

fit2$genes <- genesmd
tt2 <- topTable(fit2, coef=2, n=Inf)

par(mfrow=c(1,2), mar=c(4, 5, 2, 2))
hist(tt2$P.Value, xlab="Raw P-values", main="", las=1)
qqt(fit2$t[, 2], df=fit$df.prior+fit$df.residual, main="", pch=".", cex=3)
qqline(fit2$t[, 2], col=2, lwd=3)
abline(0, 1, lwd=3)

```



## Introducing SV as covariates

```{r DE SV}

library(sva)

mod0 <- model.matrix(~ bcr_patient_barcode, colData(prac.se.dupl))
sv <- sva(v$E, mod=design, mod0=mod0)
sv$n
vars <- length(colnames(design))
design <- cbind(design, sv$sv)
colnames(design) <- c(colnames(design)[1:vars], paste0("SV", 1:sv$n))

fit3 <- lmFit(v, design)
fit3 <- eBayes(fit3)
res3 <- decideTests(fit3, p.value=FDRcutoff)

summary(res3)


fit3$genes <- genesmd
tt4 <- topTable(fit3, coef=2, n=Inf) 
head(tt4)

sort(table(tt4$chr[tt4$adj.P.Val < FDRcutoff]), decreasing=TRUE)

par(mfrow=c(1,2), mar=c(4, 5, 2, 2))
hist(tt4$P.Value, xlab="Raw P-values", main="", las=1)
qqt(fit3$t[, 2], df=fit3$df.prior+fit3$df.residual, main="after SV", pch=".", cex=3) 
qqline(fit3$t[, 2], col=2, lwd=3)
abline(0, 1, lwd=3)

```

## Volcano plots
Volcano plots are a type of scatter-plots used to quickly identify changes in large datasets with replicate data. It contraposes fold-change (or intensity of change) and p-value (or significance of the change).

Here there will be presented 3 different models plus de base model:

- Base model represents the DEA taking into account type of tissue (tumour vs normal).

- Voom model represents the DEA taking into account the voom weight plus patients barcode (correcting for within pair variability).

- SV model represents the DEA using surrogate variable algorithms to infer the *best* correcting factors.

```{r volcano plots, fig.cap="Volcano Plots using 3 different approaches", dpi=100, fig.height= 10, fig.width=14}

par(mfrow=c(1,2))
volcanoplot(fit, coef=2, highlight=7, fit$genes$symbol, main="Base Model", las=1, cex = 0.2 )
volcanoplot(fit2, coef=2, highlight=7, fit3$genes$symbol, main="Voom model + patients", las=1, cex = 0.2)

par(mfrow=c(1,1))

volcanoplot(fit3, coef=2, highlight=7, fit3$genes$symbol, main="SV model", las=1, cex = 0.2)
abline(-log10(FDRcutoff),0)
abline(v=1)
abline(v=-1)


ggvolcano <- function(tt,FDR,hl = 5,title = element_blank(), fc = 1){
  ggplot(tt, aes(logFC, B, 
  colour = ifelse(B>-log10(FDR) & (logFC < -fc |logFC > fc) ,"red","black" ))) +
  geom_point() + 
  ggtitle(title) + geom_vline(xintercept = c(fc, -fc), col = "red")+ 
  geom_hline(yintercept =-log10(FDR), 
             col = "red") + 
  geom_text(data= tt[order(tt$B,decreasing = TRUE),][(tt$logFC < -fc | tt$logFC > fc),][1:hl,], 
  aes(logFC, B, label=symbol),angle = 25, vjust = 0) +   
  guides(colour=FALSE) + 
  theme(plot.title = element_text(size = rel(3)))
}

ggvolcano(tt4,FDRcutoff,title = "SV")
ggvolcano(tt2,FDRcutoff,title = "Voom + patient ID")
ggvolcano(tt,FDRcutoff,title = "Base Model")

```

The *best* option to consider is using SV model, as it corrects for many surrogate variables, even those that have not been detected.

It is known that SV method are stringent and some meaningful variance can be lost, however by the same reason encrease the significance of the data. ???????

From SV model can be seen that the following genomic regions are *significantly* differentially expressed:

- chr1:225

- chr2:101

- chr3:385

- chr11:11

- chrX:492



```{r}
DEgenes <- rownames(tt4)[tt4$adj.P.Val < FDRcutoff]
par(mfrow=c(1,1))
top10 <- order(fit3$lods[, 2], decreasing=TRUE)[1:3]
limma::plotMA(fit3, coef=2, status=rownames(fit3$lods) %in% DEgenes, legend=FALSE,
main="MA plot: Model SV", hl.pch=46, hl.cex=4, bg.pch=46, bg.cex=3, las=1) 
text(fit3$Amean[top10], fit3$coef[top10, 2], fit3$genes$symbol[top10], cex=0.5, pos=4)
```

Looking at the MA plot, we can see which gene or chromosome regions are significally differentially expressed (black) or not significally differentially expressed (red).

We can see that some of the chromosome mentioned before, are overexpressed in tumour samples wiht respect to normal (chr3:385, chr11:11 and chrX:492).


## Results

```{r, echo=FALSE, eval = FALSE}
library(Gviz)
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
gtrack <- GenomeAxisTrack()
txdb <- TxDb.Hsapiens.UCSC.hg19.knownGene
for (i in fit3$genes$symbol[top10][1:3]){
  gene.vars <- unlist(strsplit(i, ":"))
  
  chr <- gene.vars[1]
  coords <- unlist(strsplit(gene.vars[2], "-"))
  start <- as.numeric(coords[1])
  end <- as.numeric(coords[2])
  extrabp <- 5000
  itrack <- IdeogramTrack(genome="hg38", chromosome=chr)
  #grtrack <- GeneRegionTrack(txdb, genome="hg19", chromosome=chr,
         #                  start=start-extrabp, end=end+extrabp,
          #                 name="UCSC Known Genes", showId=TRUE, geneSymbols=TRUE)
  refseq <- UcscTrack(track="RefSeq Genes", table="refGene",
    trackType="GeneRegionTrack",chromosome=chr, genome="hg38",
    rstart="exonStarts", rends="exonEnds", gene="name", symbol="name2",
    transcript="name", strand="strand", name="RefSeq Genes",
    feature="name2",
    showId=T, from=start-extrabp, to=end+extrabp)
  plotTracks(list(gtrack, itrack, refseq), from=start-extrabp, to=end+extrabp)
}


# biomTrack <- BiomartGeneRegionTrack(genome="hg38", chromosome="chr2", start=101200000, end=101280000, name="Ensembl") 

# plotTracks(list(biomTrack,gtrack) )

```

As you can see in the below figres, our 5 top hits are:

* chr1:225 [EPHA8](http://www.genecards.org/cgi-bin/carddisp.pl?gene=EPHA8) - ephrin receptor (development of the Nervous System)

* chr2:101 [SNORD89](http://www.genecards.org/cgi-bin/carddisp.pl?gene=SNORD89) - Small Nucleolar RNA

* chr3:385 [SCN5A](http://www.genecards.org/cgi-bin/carddisp.pl?gene=SCN5A) - Sodium Channel Voltage gated V Type

* chr11:11 [TIMM8B](http://www.genecards.org/cgi-bin/carddisp.pl?gene=TIMM8B) - Translocase Inner Mithocondrial Membrane 8' Homolog

* chrX:492 [CACNA1F](http://www.genecards.org/cgi-bin/carddisp.pl?gene=CACNA1F) - Calcium Channel Voltage dependant L Type
### Functional Enrichment analysis

```{r}
DEgenes <- rownames(tt4)[tt4$adj.P.Val < FDRcutoff]
DEgenes.over <- rownames(tt4)[tt4$adj.P.Val < FDRcutoff & tt$logFC < 0]
DEgenes.under <- rownames(tt4)[tt4$adj.P.Val < FDRcutoff & tt$logFC > 0]
length(DEgenes)
library(GOstats)
params <- new("GOHyperGParams", geneIds=DEgenes,
                universeGeneIds=rownames(prac.se.dupl),
                annotation="org.Hs.eg.db", ontology="BP",
                pvalueCutoff=0.05, testDirection="over")
params.over <- new("GOHyperGParams", geneIds=DEgenes.over,
                universeGeneIds=rownames(prac.se.dupl),
                annotation="org.Hs.eg.db", ontology="BP",
                pvalueCutoff=0.05, testDirection="over")
params.under <- new("GOHyperGParams", geneIds=DEgenes.under,
                universeGeneIds=rownames(prac.se.dupl),
                annotation="org.Hs.eg.db", ontology="BP",
                pvalueCutoff=0.05, testDirection="over")


```

Now doing it conditioned by the child

```{r}
conditional(params) <- TRUE
hgOverCond <- hyperGTest(params) 
hgOverCond

```

```{r}
goresults <- GOstats::summary(hgOverCond) 
head(goresults)
dim(goresults)
```

```{r}
goresults <- goresults[ goresults$Size >= 8 & goresults$Count >= 8 , ] 
dim(goresults)
goresults <- goresults[order(goresults$Pvalue, decreasing=FALSE), ] 
head(goresults)
goresults <- goresults[order(goresults$OddsRatio, decreasing=TRUE), ] 
head(goresults)
```


```{r plotting go results}
go.plot <- ggplot(goresults, aes(Size, Count, size = OddsRatio, colour = Pvalue)) + geom_point() + xlim(5,50) + ylim(5,50) + geom_abline(intercept = 0, slope = 1) 

go.plot 
```

```{r plot go 2}
go.plot <- ggplot(goresults, aes(OddsRatio, Size, size = Count, colour = Pvalue)) + geom_point()  + ylim(0,500)

go.plot 
```

```{r plot go 3}
ggGO <- function(goresults,OR = 7.5){
  ggplot(goresults, aes(-log2(Pvalue), OddsRatio, size = Count, colour = Pvalue)) +
    geom_point() + geom_smooth(method = lm) +
    geom_text(data=goresults[goresults$OddsRatio > OR,], 
        aes(-log2(Pvalue), OddsRatio , 
            label = Term, vjust=1.5, hjust=0.5) , size=4)
}
ggGO(goresults)
```

```{r go over}
conditional(params.over) <- TRUE
hgOverCond.over <- hyperGTest(params.over) 
goresults.over <- GOstats::summary(hgOverCond.over) 
goresults.over <- goresults.over[ goresults.over$Size >= 6 & goresults.over$Count >= 6 , ] 
goresults.over <- goresults.over[order(goresults.over$OddsRatio, decreasing=TRUE), ] 
head(goresults.over, n=6)
```

```{r}
ggGO(goresults.over)
```

```{r go under}
conditional(params.under) <- TRUE
hgOverCond.under <- hyperGTest(params.under) 
goresults.under <- GOstats::summary(hgOverCond.under) 
goresults.under <- goresults.under[ goresults.under$Size >= 6 & goresults.under$Count >= 6  , ] 
goresults.under <- goresults.under[order(goresults.under$OddsRatio, decreasing=TRUE), ] 
head(goresults.under, n=6)
```

```{r}
ggGO(goresults.under)
```

```{r}
geneIDs <- geneIdsByCategory(hgOverCond)[goresults$GOBPID]
geneSYMs <- sapply(geneIDs, function(id) select(org.Hs.eg.db, columns="SYMBOL", key=id, keytype="ENTREZID")$SYMBOL)

geneSYMs <- sapply(geneSYMs, paste, collapse=", ")
goresults <- cbind(goresults, Genes=geneSYMs)
rownames(goresults) <- 1:nrow(goresults)
```

```{r printing latex table, eval=TRUE}

library(xtable)
xtab <- xtable(goresults[1:5,], align="l|c|r|r|r|r|r|p{3cm}|p{3cm}|") 
print(xtab, type="latex")

```


### GSEA

```{r loading packages, echo=FALSE}
library(GSEABase)
library(GSVAdata)
```


```{r, fig.width=10}
genesmd <- data.frame(chr=as.character(seqnames(rowRanges(prac.se.dupl))),
                         symbol=rowData(prac.se.dupl)[, 1],
                         stringsAsFactors=FALSE)


data(c2BroadSets)
c2BroadSets
length(c2BroadSets)
head(names(c2BroadSets))

c2BroadSets <- c2BroadSets[c(grep("^KEGG", names(c2BroadSets)),
               grep("^REACTOME", names(c2BroadSets)), grep("^BIOCARTA", names(c2BroadSets)))]
c2BroadSets
gsc <- GeneSetCollection(c(c2BroadSets))
gsc <- mapIdentifiers(gsc, AnnoOrEntrezIdentifier(metadata(prac.se.dupl)$annotation))
gsc

Im <- incidence(gsc)
dim(Im)
Im[1:2, 1:10]
Im <- Im[, colnames(Im) %in% rownames(prac.se.dupl)] # by a mask 
dim(Im)

prac.se.GSEA <- prac.se.dupl[colnames(Im), ]
dim(prac.se.GSEA)
prac.dge.GSEA <- prac.dgenorm[colnames(Im), ]
dim(prac.dge.GSEA)



Im <- Im[rowSums(Im) >= 5, ]
dim(Im)

tGSgenes <- tt4[match(colnames(Im), rownames(tt4)), "t"]
length(tGSgenes)
head(tGSgenes)
head(tGSgenes[order(tGSgenes, decreasing = TRUE)])
zS <- sqrt(rowSums(Im)) * (as.vector(Im %*% tGSgenes) / rowSums(Im)) #sample mean of the t statistic
length(zS)
head(zS)

qqnorm(zS)
qqline(zS)
#abline(0, 4)
```

```{r plot Gene sets, fig.width=12}
rnkGS <- sort(abs(zS), decreasing=TRUE)
head(rnkGS)

plotGS <- function(se, gs, pheno, ...) {
  l <- levels(colData(se)[, pheno])
  idxSamples1 <- colData(se)[, pheno] == l[1]
  idxSamples2 <- colData(se)[, pheno] == l[2]
  exps1 <- rowMeans(assays(se)$logCPM[gs, idxSamples1])
  exps2 <- rowMeans(assays(se)$logCPM[gs, idxSamples2])
  rng <- range(c(exps1, exps2))
  plot(exps1, exps2, pch=21, col="black", bg="black",
       xlim=rng, ylim=rng, xlab=l[1], ylab=l[2], ...)
  abline(a=0, b=1, lwd=2, col="red")
}

genesGS1 <- colnames(Im)[which(Im[names(rnkGS)[1], ] == 1)]
genesGS2 <- colnames(Im)[which(Im[names(rnkGS)[2], ] == 1)]
genesGS3 <- colnames(Im)[which(Im[names(rnkGS)[3], ] == 1)]
genesGS4 <- colnames(Im)[which(Im[names(rnkGS)[4], ] == 1)]
par(mfrow=c(1, 2), mar=c(4, 5, 3, 4))
plotGS(prac.se.GSEA, genesGS1, "type", main=names(rnkGS)[1], cex.lab=2, las=1)
plotGS(prac.se.GSEA, genesGS2, "type", main=names(rnkGS)[2], cex.lab=2, las=1)
plotGS(prac.se.GSEA, genesGS3, "type", main=names(rnkGS)[3], cex.lab=2, las=1)
plotGS(prac.se.GSEA, genesGS4, "type", main=names(rnkGS)[4], cex.lab=2, las=1)
```


## GSVA

```{r, fig.height=6, fig.width=8}
library(GSVA)

GSexpr <- gsva(assays(prac.se.dupl)$counts, gsc, rnaseq=TRUE,
               min.sz=5, max.sz=300, verbose=FALSE)$es.obs
class(GSexpr)
dim(GSexpr)


mod <- model.matrix(~ type + bcr_patient_barcode, data=colData(prac.se.dupl))
mod0 <- model.matrix(~ bcr_patient_barcode, data=colData(prac.se.dupl))
svaobj <- sva(GSexpr, mod, mod0)
modSVs <- cbind(mod, svaobj$sv)


fit <- lmFit(GSexpr, modSVs)
fit <- eBayes(fit)
tt <- topTable(fit, coef = 2, n = Inf)

tt$symbol <- rownames(tt)

DEgs <- rownames(tt[tt$adj.P.Val < 0.01, , drop=FALSE])
head(DEgs)

DEgs.over <- rownames(tt[tt$adj.P.Val < 0.01 & tt$logFC < 0 , , drop=FALSE])
DEgs.under <- rownames(tt[tt$adj.P.Val < 0.01 & tt$logFC > 0 , , drop=FALSE])
par(mfrow=c(1, 1))

tt.under <- tt[tt$logFC > 0,]

ggvolcano(tt,FDRcutoff, fc=log2(1.25), hl=10, title = "Gene Set DE - GSVA")

plot(tt$logFC, -log10(tt$P.Value), xlab="Log2 fold-change", ylab="-log10 P-value",
     pch=".", cex=5, col=grey(0.75), cex.axis=1.2, cex.lab=1.5, las=1)
points(tt[tt$adj.P.Val < 0.01, "logFC"], -log10(tt[tt$adj.P.Val < 0.01, "P.Value"]), pch=".", cex=5, col="red")
text(tt[1:2, "logFC"], -log10(tt[1:2, "P.Value"]), labels = rownames(tt[1:2,]), cex=0.5, pos=c(3,1))
text(tt.under[1:2, "logFC"], -log10(tt.under[1:2, "P.Value"]), labels = rownames(tt.under[1:2,]), cex=0.5, pos=3)
abline(h=-log10(max(tt[tt$adj.P.Val < 0.01, "P.Value"])), col=grey(0.5), lty=2)
```

```{r boxplot, fig.height=6, fig.width=10 }


boxplot(GSexpr[DEgs.over[1], ] ~ prac.se.dupl$type, main=DEgs.over[1], las=1, cex.axis=2)
boxplot(GSexpr[DEgs.over[2], ] ~ prac.se.dupl$type, main=DEgs.over[2], las=1, cex.axis=2)
boxplot(GSexpr[DEgs.under[1], ] ~ prac.se.dupl$type, main=DEgs.under[1], las=1, cex.axis=2)

```

```{r, fig.height=10, fig.width=16 }
DEgeneSets <- tt[1:30, , drop=FALSE]
dim(DEgeneSets)
GSVAsco <- GSexpr[rownames(DEgeneSets), ]

colors <- colorRampPalette( rev(brewer.pal(9, "YlOrRd")) )(255)

colorLegend <- c("darkred", "darkblue")
names(colorLegend) <- c("tumor", "normal")
sample.color.map <- colorLegend[colData(prac.se.dupl)$type]
names(sample.color.map) <- colnames(GSVAsco)
sampleClustering <- hclust(as.dist(1-cor(GSVAsco, method="spearman")), method="complete")
geneSetClustering <- hclust(as.dist(1-cor(t(GSVAsco), method="pearson")), method="complete")
heatmap(GSVAsco, ColSideColors=sample.color.map, xlab="samples",
        ylab="Gene sets and pathways", margins=c(2, 20),
        labRow=substr(gsub("_", " ", gsub("^KEGG_|^REACTOME_|^BIOCARTA_", "", rownames(GSVAsco))), 1, 35),
        labCol="", scale="row",
        Colv=as.dendrogram(sampleClustering), Rowv=as.dendrogram(geneSetClustering), col=colors)
legend("topleft", names(colorLegend), fill=colorLegend, inset=0.01, bg="white")

```

# Session Info 

```{r session info}
sessionInfo()
```


# References
