---
title: "PRAC Paired Design"
author: "David Mas"
date: "1 de junio de 2016"
output:
  html_document:
    mathjax: local
    self_contained: false
---


```{r setup, cache=FALSE, echo=FALSE, message=FALSE, warning=FALSE, tidy=FALSE}
library(knitr)
opts_chunk$set(cache=T, fig.align='center', dpi=100, message=FALSE)

```


```{r loading packages and reading data, echo=FALSE}
library("SummarizedExperiment")
library("edgeR")
library("geneplotter")
library("ggplot2")
library("sva")
old.par <- par(mar = c(0, 0, 0, 0)) # to get CLI 

prac.se <- readRDS("data/sePRAD.rds")

```

# Filtering and Data description

```{r original dataset}

table(prac.se$type)

```

In this design we are going to select only the paired samples. Therefore, each patient have to be the source for an assigned as tumor sample and a assigned normal sample. 

```{r filtering for the paired data}
# Subset the paired samples
## we build a table from the bcr barcode occurences in our dataset
n_occur <- data.frame(table(colData(prac.se)$bcr_patient_barcode))

## we filter out the na patients (having a na in the barcode)
## and the ones that occur only 1 time (they are unique)
prac.se.dupl <- prac.se[,colData(prac.se)$bcr_patient_barcode
                 %in% n_occur$Var1[n_occur$Freq > 1] & 
                 !is.na(colData(prac.se)$bcr_patient_barcode)]
## See the results.
table(colData(prac.se.dupl)$type)

```

Most of the Normal samples had a paired tumor sample but 2. 

```{r Normalizing}

prac.dge <- DGEList(counts = assays(prac.se.dupl)$counts, genes = mcols(prac.se.dupl), group = prac.se.dupl$type)
prac.dgenorm <- calcNormFactors(prac.dge)
assays(prac.se.dupl)$logCPM <- cpm(prac.dgenorm, log=TRUE, prior.count=3.5)
assays(prac.se.dupl)$logCPM[1:5, 1:5]
```

To avoid errors or misleading data in the future we set up a threshold of the gene expression to remove the low expressed genes. 

```{r filter genes}
mask <- rowMeans(assays(prac.se.dupl)$logCPM) > 1
prac.se.dupl <- prac.se.dupl[mask, ]
prac.dgenorm <- prac.dgenorm[mask, ]
```

We explore the data to check weather there exists some kind of batch effect or an undesired source of variation.

```{r lib size plots}
libsize <- data.frame(libsize = prac.dgenorm$sample$lib.size/1e6, type = prac.se.dupl$type, samples = rownames(prac.dgenorm$samples) )
ggplot(libsize, aes(x=reorder(samples, libsize), y=libsize, fill=type)) +  theme(plot.background = element_blank(), axis.title.x=element_blank(), axis.text.x = element_blank()) + geom_bar(stat = "identity") 

```


The potencial source of batch effect is the TSS so we check for that.

```{r batch effect, echo = FALSE}

tss <- substr(colnames(prac.se.dupl), 6, 7)
table(tss)
center <- substr(colnames(prac.se.dupl), 27, 28)
table(center)
plate <- substr(colnames(prac.se.dupl), 22, 25)
table(plate)
portionanalyte <- substr(colnames(prac.se.dupl), 18, 20)
table(portionanalyte)
samplevial <- substr(colnames(prac.se.dupl), 14, 16)
table(samplevial)

table(data.frame(TYPE=prac.se.dupl$type, TSS=tss))


par(old.par)
logCPM <- assays(prac.se.dupl)$logCPM
d <- as.dist(1-cor(logCPM, method="spearman"))
sampleClustering <- hclust(d)
batch <- as.integer(factor(tss))
colors <- palette()
palette(c(colors,"darkorange1","darkmagenta","darkolivegreen","red3","seagreen","pink1","yellow3","sienna1"))
sampleDendrogram <- as.dendrogram(sampleClustering, hang=0.1)
names(batch) <- colnames(prac.se.dupl)
outcome <- paste(substr(colnames(prac.se.dupl), 9, 12), as.character(prac.se.dupl$type), sep="-")
names(outcome) <- colnames(prac.se.dupl)
sampleDendrogram <- dendrapply(sampleDendrogram,
                               function(x, batch, labels) {
                                 if (is.leaf(x)) {
                                   attr(x, "nodePar") <- list(lab.col=as.vector(batch[attr(x, "label")]))
                                   attr(x, "label") <- as.vector(labels[attr(x, "label")])
                                 }
                                 x
                               }, batch, outcome)

plot(sampleDendrogram, main="Hierarchical clustering of samples")
legend("topright", paste("Batch", sort(unique(batch)), levels(factor(tss))), fill=sort(unique(batch)))

```


```{r MDS, echo=FALSE}
par(mfrow=c(1, 1), mar=c(0,0,0,0))
plotMDS(prac.dgenorm, labels=outcome, col=batch)
legend("topright", paste("Batch", sort(unique(batch)), levels(factor(tss))),
       fill=sort(unique(batch)), inset=0.05)
type <- as.integer(factor(prac.se.dupl$type))
plotMDS(prac.dgenorm, pch = 19, col=type)
legend("topright", paste("Type", sort(unique(type)), levels(factor(prac.se.dupl$type))),
       fill=sort(unique(type)), inset=0.05)
```


```{r filter those ind}
GetID <- function(number, se){
  id <- grep(number,rownames(colData(se)))
  return(rownames(colData(se))[id])
}
number.id <- c("7737","7745","7738","7211","7747","6356","6362","5769") 
pruning.vec <- as.character(sapply(number.id, GetID, prac.se.dupl))
prac.dgenorm <- prac.dgenorm[,
                   !rownames(prac.dgenorm$samples) %in% pruning.vec]
table(prac.dgenorm$samples$group)
prac.se.dupl <- prac.se.dupl[,rownames(prac.dgenorm$samples)]
table(colData(prac.se.dupl)$type)
```

```{r MDS2, echo=FALSE}
tss <- substr(colnames(prac.se.dupl), 6, 7)
batch <- as.integer(factor(tss))
names(batch) <- colnames(prac.se.dupl)
outcome <- paste(substr(colnames(prac.se.dupl), 9, 12), as.character(prac.se.dupl$type), sep="-")


par(mfrow=c(1, 1), mar=c(0,0,0,0))
plotMDS(prac.dgenorm, labels=outcome, col=batch)
legend("topright", paste("Batch", sort(unique(batch)), levels(factor(tss))),
       fill=sort(unique(batch)), inset=0.05)
type <- as.integer(factor(prac.se.dupl$type))
plotMDS(prac.dgenorm, pch = 19, col=type)
legend("topright", paste("Type", sort(unique(type)), levels(factor(prac.se.dupl$type))),
       fill=sort(unique(type)), inset=0.05)
```

# Differential Expression Analysis

```{r DE1}
colData(prac.se.dupl)$bcr_patient_barcode <- droplevels(colData(prac.se.dupl)$bcr_patient_barcode)
design <- model.matrix(~ type , data=colData(prac.se.dupl))
head(design)
tail(design)
fit <- lmFit(assays(prac.se.dupl)$logCPM, design)
fit <- eBayes(fit)
FDRcutoff <- 0.01
res <- decideTests(fit, p.value=FDRcutoff)
summary(res)
genesmd <- data.frame(chr=as.character(seqnames(rowRanges(prac.se.dupl))),
                      symbol=as.character(rowRanges(prac.se.dupl)[, 1]),
                      stringsAsFactors=FALSE)
fit$genes <- genesmd
tt <- topTable(fit, coef=2, n=Inf)
head(tt)
sort(table(tt$chr[tt$adj.P.Val < FDRcutoff]), decreasing=TRUE)


par(mfrow=c(1,2), mar=c(4, 5, 2, 2))
hist(tt$P.Value, xlab="Raw P-values", main="", las=1)
qqt(fit$t[, 2], df=fit$df.prior+fit$df.residual, main="", pch=".", cex=3)
qqline(fit$t[, 2], col=2, lwd=3)
abline(0, 1, lwd=3)

```

In black we can see the Normal distribution ( \(\sigma = 1 \) ) and the red is the `qqline` extracted from the distribution itself.

Right now we have a real true candidate overexpressed in tumor at the right top. The other genes, although with a high variability seem to follow a normal distribution of \(\sigma = 4 \).

## Improving the DEA with voom

```{r DE voom}
par(mfrow=c(1,1))
v <- voom(prac.dgenorm, design, plot=TRUE)

# redo the process with the voom weight

fit2 <- lmFit(v, design)
fit2 <- eBayes(fit2)
res2 <- decideTests(fit2, p.value=FDRcutoff)

fit2$genes <- genesmd
tt2 <- topTable(fit2, coef=2, n=Inf)

par(mfrow=c(1,2), mar=c(4, 5, 2, 2))
hist(tt2$P.Value, xlab="Raw P-values", main="", las=1)
qqt(fit2$t[, 2], df=fit$df.prior+fit$df.residual, main="", pch=".", cex=3)
qqline(fit2$t[, 2], col=2, lwd=3)
abline(0, 1, lwd=3)

```


## Introducing TSS as covariate

```{r DE TSS}

tss <- factor(tss)
design <- model.matrix(~ factor(type) + tss , data=colData(prac.se.dupl))


fit3 <- lmFit(v, design)
fit3 <- eBayes(fit3)

fit3$genes <- genesmd

tt3 <- topTable(fit3, coef=2, n=Inf)

par(mfrow=c(1,2), mar=c(4, 5, 2, 2))
hist(tt3$P.Value, xlab="Raw P-values", main="", las=1)
qqt(fit3$t[, 2], df=fit3$df.prior+fit3$df.residual, main="", pch=".", cex=3)
qqline(fit3$t[, 2], col=2, lwd=3)
abline(0, 1, lwd=3)

```

## Introducing SV as covariates

```{r DE SV}

library(sva)

mod0 <- model.matrix(~ 1, colData(prac.se.dupl))
sv <- sva(v$E, mod=design, mod0=mod0)
sv$n
vars <- length(colnames(design))
design <- cbind(design, sv$sv)
colnames(design) <- c(colnames(design)[1:vars], paste0("SV", 1:sv$n))

fit4 <- lmFit(v, design)
fit4 <- eBayes(fit4)
res4 <- decideTests(fit4, p.value=FDRcutoff)

summary(res4)


fit4$genes <- genesmd
tt4 <- topTable(fit4, coef=2, n=Inf) 
head(tt4)

sort(table(tt4$chr[tt4$adj.P.Val < FDRcutoff]), decreasing=TRUE)

par(mfrow=c(1,2), mar=c(4, 5, 2, 2))
hist(tt4$P.Value, xlab="Raw P-values", main="", las=1)
qqt(fit4$t[, 2], df=fit4$df.prior+fit4$df.residual, main="after SV", pch=".", cex=3) 
qqline(fit4$t[, 2], col=2, lwd=3)
abline(0, 1, lwd=3)

```

## Volcano plots

```{r volcano plots, fig.cap="Volcano Plots using 3 different approaches", dpi=100, fig.height= 10, fig.width=14}
par(mfrow=c(2,2))
volcanoplot(fit, coef=2, highlight=7, fit$genes$symbol, main="Base Model", las=1)
volcanoplot(fit2, coef=2, highlight=7, fit4$genes$symbol, main="Model voom", las=1)
volcanoplot(fit3, coef=2, highlight=7, fit4$genes$symbol, main="Model voom + TSS", las=1)
volcanoplot(fit4, coef=2, highlight=7, fit4$genes$symbol, main="Model voom + TSS + SV", las=1)
```



```{r}
DEgenes <- rownames(tt4)[tt4$adj.P.Val < FDRcutoff]
par(mfrow=c(1,1))
top10 <- order(fit4$lods[, 2], decreasing=TRUE)[1:10]
limma::plotMA(fit4, coef=2, status=rownames(fit4$lods) %in% DEgenes, legend=FALSE,
main="Model SV", hl.pch=46, hl.cex=4, bg.pch=46, bg.cex=3, las=1) 
text(fit4$Amean[top10], fit4$coef[top10, 2], fit4$genes$symbol[top10], cex=0.5, pos=4)
```


## Results

```{r, echo=FALSE, eval = FALSE}
library(Gviz)
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
gtrack <- GenomeAxisTrack()
txdb <- TxDb.Hsapiens.UCSC.hg19.knownGene
for (i in fit4$genes$symbol[top10][1:3]){
  gene.vars <- unlist(strsplit(i, ":"))
  
  chr <- gene.vars[1]
  coords <- unlist(strsplit(gene.vars[2], "-"))
  start <- as.numeric(coords[1])
  end <- as.numeric(coords[2])
  extrabp <- 5000
  itrack <- IdeogramTrack(genome="hg38", chromosome=chr)
  #grtrack <- GeneRegionTrack(txdb, genome="hg19", chromosome=chr,
         #                  start=start-extrabp, end=end+extrabp,
          #                 name="UCSC Known Genes", showId=TRUE, geneSymbols=TRUE)
  refseq <- UcscTrack(track="RefSeq Genes", table="refGene",
    trackType="GeneRegionTrack",chromosome=chr, genome="hg38",
    rstart="exonStarts", rends="exonEnds", gene="name", symbol="name2",
    transcript="name", strand="strand", name="RefSeq Genes",
    feature="name2",
    showId=T, from=start-extrabp, to=end+extrabp)
  plotTracks(list(gtrack, itrack, refseq), from=start-extrabp, to=end+extrabp)
}


# biomTrack <- BiomartGeneRegionTrack(genome="hg38", chromosome="chr2", start=101200000, end=101280000, name="Ensembl") 

# plotTracks(list(biomTrack,gtrack) )

```

As you can see in the below figres, our 3 top hits are:

* [SNORD89](http://www.genecards.org/cgi-bin/carddisp.pl?gene=SNORD89) - Small Nucleolar RNA
* [EPHA8](http://www.genecards.org/cgi-bin/carddisp.pl?gene=EPHA8) - ephrin receptor (development of the Nervous System)
* [EFHD1](http://www.genecards.org/cgi-bin/carddisp.pl?gene=EFHD1) - EF-hand domain-containing protein related to calcium ion binding (neuronal differentiation)



### Functional Enrichment analysis

```{r}
DEgenes <- rownames(tt4)[tt$adj.P.Val < 0.001]
length(DEgenes)
library(GOstats)
params <- new("GOHyperGParams", geneIds=DEgenes,
                universeGeneIds=rownames(prac.se.dupl),
                annotation="org.Hs.eg.db", ontology="BP",
                pvalueCutoff=0.05, testDirection="over")
hgOver <- hyperGTest(params) 

hgOver

head(summary(hgOver))

head(geneCounts(hgOver))

head(universeCounts(hgOver))
head(pvalues(hgOver))
```

Now doing it conditioned by the child

```{r}
conditional(params) <- TRUE
hgOverCond <- hyperGTest(params) 
hgOverCond
```

```{r}
goresults <- summary(hgOverCond) 
head(goresults)
```

```{r}
goresults <- goresults[goresults$Size >= 7 & goresults$Count >= 7, ] 
goresults <- goresults[order(goresults$OddsRatio, decreasing=TRUE), ] 
head(goresults)
```

```{r}
geneIDs <- geneIdsByCategory(hgOverCond)[goresults$GOBPID]
geneSYMs <- sapply(geneIDs, function(id) select(org.Hs.eg.db, columns="SYMBOL", key=id, keytype="ENTREZID")$SYMBOL)

geneSYMs <- sapply(geneSYMs, paste, collapse=", ")
goresults <- cbind(goresults, Genes=geneSYMs)
rownames(goresults) <- 1:nrow(goresults)


#library(xtable)
#xtab <- xtable(goresults[1:5,], align="l|c|r|r|r|r|r|p{3cm}|p{3cm}|") 
#print(xtab, file="goresults.tex", type="latex")

```


### GSEA

```{r, fig.width=14}
genesmd <- data.frame(chr=as.character(seqnames(rowRanges(prac.se.dupl))),
                         symbol=rowData(prac.se.dupl)[, 1],
                         stringsAsFactors=FALSE)

library(GSEABase)
library(GSVAdata)
data(c2BroadSets)
c2BroadSets
length(c2BroadSets)
head(names(c2BroadSets))

c2BroadSets <- c2BroadSets[c(grep("^KEGG", names(c2BroadSets)),
               grep("^REACTOME", names(c2BroadSets)), grep("^BIOCARTA", names(c2BroadSets)))]
c2BroadSets
gsc <- GeneSetCollection(c(c2BroadSets))
gsc <- mapIdentifiers(gsc, AnnoOrEntrezIdentifier(metadata(prac.se.dupl)$annotation))
gsc

Im <- incidence(gsc)
dim(Im)
Im[1:2, 1:10]
Im <- Im[, colnames(Im) %in% rownames(prac.se.dupl)] # by a mask 
dim(Im)

Im <- Im[, colnames(Im) %in% rownames(prac.se.dupl)] # by a mask
dim(Im)

prac.se.GSEA <- prac.se.dupl[colnames(Im), ]
dim(prac.se.GSEA)
prac.dge.GSEA <- prac.dgenorm[colnames(Im), ]
dim(prac.dge.GSEA)

library(limma)
library(sva)


# Im <- Im[rowSums(Im) >= 5, ]
dim(Im)

tGSgenes <- tt4[match(colnames(Im), rownames(tt4)), "t"]
length(tGSgenes)
head(tGSgenes)

zS <- sqrt(rowSums(Im)) * (as.vector(Im %*% tGSgenes) / rowSums(Im)) #sample mean of the t statistic
length(zS)
head(zS)

qqnorm(zS)
abline(0, 4)

rnkGS <- sort(abs(zS), decreasing=TRUE)
head(rnkGS)

plotGS <- function(se, gs, pheno, ...) {
  l <- levels(colData(se)[, pheno])
  idxSamples1 <- colData(se)[, pheno] == l[1]
  idxSamples2 <- colData(se)[, pheno] == l[2]
  exps1 <- rowMeans(assays(se)$logCPM[gs, idxSamples1])
  exps2 <- rowMeans(assays(se)$logCPM[gs, idxSamples2])
  rng <- range(c(exps1, exps2))
  plot(exps1, exps2, pch=21, col="black", bg="black",
       xlim=rng, ylim=rng, xlab=l[1], ylab=l[2], ...)
  abline(a=0, b=1, lwd=2, col="red")
}

genesGS1 <- colnames(Im)[which(Im[names(rnkGS)[1], ] == 1)]
genesGS2 <- colnames(Im)[which(Im[names(rnkGS)[2], ] == 1)]
genesGS3 <- colnames(Im)[which(Im[names(rnkGS)[3], ] == 1)]
genesGS4 <- colnames(Im)[which(Im[names(rnkGS)[4], ] == 1)]
par(mfrow=c(1, 2), mar=c(4, 5, 3, 4))
plotGS(prac.se.GSEA, genesGS1, "type", main=names(rnkGS)[1], cex.lab=2, las=1)
plotGS(prac.se.GSEA, genesGS2, "type", main=names(rnkGS)[2], cex.lab=2, las=1)
plotGS(prac.se.GSEA, genesGS3, "type", main=names(rnkGS)[3], cex.lab=2, las=1)
plotGS(prac.se.GSEA, genesGS4, "type", main=names(rnkGS)[4], cex.lab=2, las=1)
```


## GSVA

```{r}
library(GSVA)

GSexpr <- gsva(assays(prac.se.GSEA)$counts, gsc, rnaseq=TRUE,
               min.sz=5, max.sz=300, verbose=FALSE)$es.obs
class(GSexpr)
dim(GSexpr)


mod <- model.matrix(~ type, data=colData(prac.se.dupl))
mod0 <- model.matrix(~ 1, data=colData(prac.se.dupl))
svaobj <- sva(GSexpr, mod, mod0)
modSVs <- cbind(mod, svaobj$sv)


fit <- lmFit(GSexpr, modSVs)
fit <- eBayes(fit)
tt <- topTable(fit, coef = 2, n = Inf)
DEgs <- rownames(tt[tt$adj.P.Val < 0.01, , drop=FALSE])
head(DEgs)



par(mfrow=c(1, 1))

#boxplot(GSexpr["XiE", ] ~ lclse$gender, main="XiE", las=1, cex.axis=2)

plot(tt$logFC, -log10(tt$P.Value), xlab="Log2 fold-change", ylab="-log10 P-value",
     pch=".", cex=5, col=grey(0.75), cex.axis=1.2, cex.lab=1.5, las=1)
points(tt[tt$adj.P.Val < 0.01, "logFC"], -log10(tt[tt$adj.P.Val < 0.01, "P.Value"]), pch=".", cex=5, col="red")
text(tt[tt$adj.P.Val < 1e-20, "logFC"], -log10(tt[tt$adj.P.Val < 1e-20, "P.Value"]), labels = rownames(tt[tt$adj.P.Val < 1e-20,]), cex=0.7, pos=4)
abline(h=-log10(max(tt[tt$adj.P.Val < 0.01, "P.Value"])), col=grey(0.5), lty=2)

boxplot(GSexpr[DEgs[1], ] ~ prac.se.GSEA$type, main=DEgs[1], las=1, cex.axis=2)

boxplot(GSexpr[DEgs[2], ] ~ prac.se.GSEA$type, main=DEgs[2], las=1, cex.axis=2)


boxplot(GSexpr[DEgs[5], ] ~ prac.se.GSEA$type, main=DEgs[5], las=1, cex.axis=2)
```

# Session Info 

```{r session info}
sessionInfo()
```


