---
title: 'Analysis of a TCGA RNA-seq data set on Prostate Adenocarcinoma '
author: "Joan Martí, David Mas, Adrià Auladell"
date: "29 de abril de 2016"
output: html_document
---

## Introduction

[Prostate cancer](https://en.wikipedia.org/wiki/Prostate_cancer)
is a disease of the prostate, a walnut-size gland in the male
reproductive system.  Nearly all prostate cancer is prostate adenocarcinoma.
 Prostate cancer is graded based on its Gleason score, which is how the cells
 look under a microscope and ranges from two to ten. A low Gleason score means
 that the cancer tissue is similar to normal cells and unlikely to spread. A
 high Gleason score means that the cancer cells are very different from normal
 cells and are likely to spread. For patients whose cancer has spread, their
 survival time is usually one to three years. It was estimated that for 2011,
 240,890 men would be diagnosed and 33,720 would die from prostate cancer.

The Cancer Genome Atlas already studied this dataset for a wide characterization of the prostate adenocarcinoma subtypes. TCGA established the following points:

* 74% of all tumors being assignable to one of seven molecular classes based on distinct oncogenic drivers: 
    - fusions involving (1) **ERG**, (2) **ETV1**, (3) **ETV4**, or (4) **FLI1** (46%, 8%, 4%, and 1%, respectively)
    - mutations in (5) **SPOP** or (6) **FOXA1**; or (7) **IDH1**. (11%, 3%, and 1%, respectively).
    
* 25% of the prostate cancers had a presumed actionable lesion in the PI3K or MAPK signaling pathways, and DNA repair genes were inactivated in 19%.

You can find the full article [here](http://dx.doi.org/10.1016/j.cell.2015.10.025).

## Data import

### Package Installation and Import
First, we install the packages using BiocInstaller:

```{r install, eval=FALSE}
source("http://www.bioconductor.org/biocLite.R")
library(BiocInstaller)
biocLite(c("edgeR","SummarizedExperiment","geneplotter","sva"))
install.packages(ggplot2)
```




```{r loading, message=FALSE}
library("SummarizedExperiment")
library("edgeR")
library("geneplotter")
library("ggplot2")
library("sva")
old.par <- par(mar = c(0, 0, 0, 0)) # to get CLI 
```

### Data Reading
After installing and loading the packages, we use them to import the [dataset](http://functionalgenomics.upf.edu/courses/IEO/projects/datasets/sePRAD.rds). This dataset is provided by our professor [Robert Castelo](mailto:robert.castelo@upf.edu). 

The data is contained in a rds object and when reading it by the function `readRDS` we extract the `Summarized Experiment`object that contains the data we are going to use. This class is from the `SummarizedExperiment` package loaded above. 

```{r reading}
prac.se <- readRDS("data/sePRAD.rds")
prac.se
```

### Nomenclature

In the this and following reports we are going to try naming variables and objects using a nomenclature. We believe this nomenclature follows the Google's R Style Guidelines. As recomended by this guidelines, different parts of the variable name are separated by points. 

* The first part of the variable defines what it contains in a clear as possible. 
* Second part defines the `class` of the object. This is missed when the object have a common class (such as character vectors) or it is a temporary variable. 
* Third pard defines the filters applied to the original object. 

Based on this nomenclature, below you can find variables like `<name>.<type_var>.<other>`:

```
prac.se 
# where prac is the item it represent (cancer type) and
# se is the class (summarized Experiment)
prac.se.unique 
# where a third labeled is applied in order to note that the data have 
# been filtred. 
```
We also try to mantain our code chunks below 80 chrs.

### Exploring the DataSet

As defined in the `SummarizedExperiment` documentation there are 3 data types in a SE. The **Column data** contains info about the samples used in the experiment and it is refered as column data because represent the columns of the original expression table. Then we have the **metadata** that brings us information about the assays that were performed. Finally we got the **row data** that contains the gene information. This is called row data because in the expression table genes are in the rows. 

#### Data Size
```{r dim}
dim(colData(prac.se))
```

Our dataset presents 554 different samples with 549 clinical variables analyzed. 


#### Column Data

In order to acces the column data from our `SE`object we can use its own implemented function, `colData()`.

```{r data}
head(names(colData(prac.se)))
colData(prac.se)[1:5, 1:5]
mcols(colData(prac.se), use.names=TRUE)
```

By observing the clinical variables, from the 549 present some have information collected and others present NA values. Our interest is in the informative variables.Some of them are useful to study the data, like bcr\_patient\_barcode, race, ethnicity, age\_at\_initial\_pathologic\_diagnosis...There are variables related directly with prostate cancer, like gleason score, the measurement of the development of the adenocarcinoma.

```{r exploring some variables, fig.align='center', echo=FALSE, message=FALSE}
sampleinfo.df <- as.data.frame(colData(prac.se))
require(gridExtra)
plot1 <- ggplot(sampleinfo.df, aes(x=vital_status, fill =gleason_score)) + geom_bar()
plot2 <- ggplot(sampleinfo.df, aes(x=race, fill =gleason_score)) + geom_bar() + theme(axis.text.x = element_text(angle = 25, hjust = 1))
grid.arrange(plot1, plot2, ncol=2)
```

Looking at the metadata, we observe the common structure followed in the datasets of the TCGA project. The first column is the clinical variable abreviated, the second a explanation of the variable and the last one contains the  [CDEID](https://www.nlm.nih.gov/cde/glossary.html#cdedefinition ) code to obtain more information in the [CDE website](https://www.nlm.nih.gov/cde). 

As an example we can see in the figures, there are clinical variables and sample variables giving information about the sampels. 

#### Tumor / Normal Samples
```{r table types}
table(prac.se$type)

```

The dataset presents 554 samples. There are 52 labeled as normal samples and 504 labeled as tumor samples. The first ones are not from healthy individuals but from affected ones. This mean that normal samples do not have tumor phenotype but the same individual background than the tumor samples. In fact, most of the normal samples (up to 50) have a paired sample (2 samples from the same individual) in the tumor side of the table. We can check this information using the BCR patient barcode that identifies individuals. 


#### Row Data

We can get information about the genes using the following. 

```{r row data}
rowRanges(prac.se)
```

For each gene we have present the chromosome, the range of position, the strand, its symbol, the length of the transcript and the GC content. 
```{r gc content hist, fig.align='center', echo=FALSE}
geneinfo.df <- as.data.frame(mcols(prac.se))
plot1 <- ggplot(geneinfo.df, aes(x=txlen)) + geom_histogram(bins = 50)
plot2 <- ggplot(geneinfo.df, aes(x=txgc)) + geom_histogram(bins = 50)
grid.arrange(plot1, plot2, ncol=2)
```

#### Metadata
From the metadata we can check that we have the current version of the data. 
```{r metadata}
metadata(prac.se)$objectCreationDate #the result should be [1] "2016-04-25"
```

## Subsetting the Data

In order to reduce possible batch effects and reduce the computational costs of the analysis we have decided to subset our data. We will use several approaches in order to do perform this action. Finally, we will apply one or other subset if we want to avoid the disadvantages of a defined strategy.

### **Paired Subsetting** 

In the paired subsetting we try to get only the patients (based on the bcr barcode) that have a normal sample and a tumor one. This approach is suitable to distinguish exactly the gene expression changes that drive the tumour. However, it also drives some disadvantages as we have been taugh in class.

* As seen in the resulting numbers, we have filter out 454 patients that were not paired. We have same number of normal samples and tumor samples (50) because for each individual we have 2 samples, each one in a different group.* 

```{r subset}
# Subset the paired samples
## we build a table from the bcr barcode occurences in our dataset
n_occur <- data.frame(table(colData(prac.se)$bcr_patient_barcode))

## we filter out the na patients (having a na in the barcode)
## and the ones that occur only 1 time (they are unique)
prac.se.dupl <- prac.se[,colData(prac.se)$bcr_patient_barcode
                 %in% n_occur$Var1[n_occur$Freq > 1] & 
                 !is.na(colData(prac.se)$bcr_patient_barcode)]
## See the results.
table(colData(prac.se.dupl)$type)
```
 
### **Non-Paired Subsetting** 

With this strategy we want to discard all the paired data in order to obtain only unique individuals. Our aim is to only get the normal samples of the replicated individuals. As the initial filter strategy get rid of most of the normal samples and kept the tumor ones we had to adapt it and generate a more complex filter as you can see below. 

*The replicated samples are 2 samples extracted from the same individual in 2 different conditions (tumor and normal). We want to distinguish general variation that is associated to one or the other subset and introducing 2 samples for the same individual could include batch effect in our samples.*

*In the resuting table we can see that all the normal samples remain meanwhile we have filter out 67 tumor samples that contained `NA`values in the bcr barcode field or that they were extracted from the same individuals that provided the normal samples.*
    
```{r nonpaired subset}
# Subset the non-paired 
prac.se.unique <- prac.se[, colData(prac.se)$bcr_patient_barcode 
                             %in% n_occur$Var1[n_occur$Freq == 1] & 
                                !is.na(colData(prac.se)$bcr_patient_barcode) |
                                (colData(prac.se)$bcr_patient_barcode %in% 
                                n_occur$Var1[n_occur$Freq > 1] & 
                                colData(prac.se)$type == "normal" & 
                                !is.na(colData(prac.se)$bcr_patient_barcode))]
table(colData(prac.se.unique)$type)

```
  

## Quality data control

Having understood completely our dataset, to perform efficiently quality assesment and normalization of the data, we have to convert our values to counts per million values (CPM). In order to do that, the package edgeR is used, creating a 'DGEList Object'. 
Digital Gene Expression data class (DGE) has been implemented for storing read counts and associated information from digital gene expression or sequencing technologies. The user specifies the counts, the samples, and optional components include the genes and the groups.

```{r dge, warning= FALSE}
prac.dge <- DGEList(counts = assays(prac.se)$counts, genes = mcols(prac.se), group = prac.se$type)
head(prac.dge$samples)
```

Finally, from each CPM value we calculate the $\log_2$ measure and we include it in our dataset in order to use it in the following comparisons/normalizations. 

```{r log2}
assays(prac.se)$logCPM <- cpm(prac.dge, log=TRUE, prior.count=3.5)
assays(prac.se)$logCPM[1:5, 1:5]
```

### Filtering by library size
Now we subset our dataset by library size, which is a measure of sequencing deepness or how robust are the RNA-seqs by samples. We renormalize the data with the new `SE`filtred element. 

```{r normalitzation of the filtred dataset, warning= FALSE}
prac.dge.unique <- DGEList(counts = assays(prac.se.unique)$counts, genes = mcols(prac.se.unique), group = prac.se.unique$type)
```

```{r library size all after filter unique, echo=FALSE, fig.align='center', fig.cap='Fig. 1: Library Size per sample in the  non-paired design dataset. '}

ord <- order(prac.dge.unique$samples$lib.size/1e6)
barplot((prac.dge.unique$sample$lib.size/1e06)[ord], las=1, ylab="Millions of reads",
        xlab="Samples", col=c("blue", "red")[(prac.se$type[ord] == "tumor") + 1])
legend("topleft", c("tumor", "normal"), fill=c("red", "blue"), inset=0.01)

```

In the fig. 1 , we cannot distinguish the different samples due the large number of samples. We will filter in a high treshold of sequencing depth to recude the number of samples erasing the ones with less sequencing quality. 


```{r lib.unique}
# library size, non-paired samples
prac.dge.unique.filtlib <- prac.dge.unique[,(prac.dge.unique$samples$lib.size/1e6) > 40 ] # 50e06 threshold value
table(prac.dge.unique.filtlib$samples$group)
```
<!--- Adri START --->


Since we still have a lot of samples, we look at the distribution of the Tissue Soucre Site (TSS) for selecting the batches with a good distribution of normal/tumor samples in order to obtain a robust dataset.


```{r filter}
# not filtering by TSS
prac.se.sub <- prac.se.unique[,rownames(prac.dge.unique.filtlib$samples)]

# filter by TSS block
tss <- substr(colnames(prac.se.sub), 6, 7)
tss_df <- data.frame(TYPE=prac.se.sub$type, TSS=tss)
table(data.frame(TYPE=prac.se.sub$type, TSS=tss))
```


We can see some TSS presenting only tumor samples with a low number. These type of samples can be a potential batch factor, so we choose a selection of balanced samples (for example CH, EJ, G9 and HC).

```{r selec}
selection  <- c('CH', 'EJ', 'G9', 'HC')
prac.dge.unique.filtlib <- prac.dge.unique.filtlib[,tss_df$TSS %in% selection]
prac.se.sub <- prac.se.unique[,rownames(prac.dge.unique.filtlib$samples)]
table(prac.dge.unique.filtlib$samples$group)
```

 <!--- Adri END --->

```{r figure library size, echo=FALSE, fig.align='center', fig.cap='Fig. 2: Density sample distribution using already filtred data by sequencing depth.'}

ord.unique <- order(prac.dge.unique.filtlib$samples$lib.size/1e6)

libsize <- data.frame(libsize = prac.dge.unique.filtlib$sample$lib.size/1e6, type = prac.se.sub$type, samples = rownames(prac.dge.unique.filtlib$samples) )

summary(libsize)


ggplot(libsize) +
  geom_density(aes(x=libsize, fill=type), alpha=0.5) +
  ylab("Density\n") + xlab("\nMillions of Reads") +
  theme_bw()

```

```{r barplot libsize samples filtred 2, echo=FALSE, fig.align='center',fig.cap="Fig. 3: Barplot of the final selected samples with its library size."}


ggplot(libsize, aes(x=reorder(samples, libsize), y=libsize, fill=type)) + geom_bar(stat = "identity") + theme(axis.text.x=element_blank(), axis.title.x=element_blank())

#barplot((prac.dge.unique.filtlib$sample$lib.size/1e06)[ord.unique], las=1, ylab="Millions of reads",
#        xlab="Samples", col=c("blue", "red")[(prac.dge.unique.filtlib$sample$group[ord.unique] == "tumor") + 1])

#legend("topleft", c("tumor", "normal"), fill=c("red", "blue"), inset=0.01)
```

After filtering, we obtain a dataset of 48 normal samples and 122 tumor (fig. 2 and 3). We introduce it again in the prac.se dataset to obtain the final filtered version.
Bear in mind that it seems there is an outlayer, a normal sample with an unusual or unexpected good quality. The converage is unusually high but since it is theorically a good attribute, we decided to keep it. 

### Sample expression distribution

Once we have our filtered dataset, we observe the density logCPM distribution of the tumor and normal samples separatedly. It will give us an impression of the possible problems of the sampling, since we expect to have a similar distribution of the samples between tumor / normal. 

```{r Multydensity, fig.align='center', fig.cap='Fig 4: Multidensity plot representing the clusters of the non expressed genes at the left and the expressed genes at the right.', echo=FALSE, fig.width=12}
MDS_normal <- prac.dge.unique.filtlib[,prac.dge.unique.filtlib$samples$group == 'normal']
MDS_tumor <- prac.dge.unique.filtlib[,prac.dge.unique.filtlib$samples$group == 'tumor']

logCPM.MDS_normal <- cpm(MDS_normal, log = TRUE, prior.count = 3.25)
logCPM.MDS_tumor <- cpm(MDS_tumor, log = TRUE, prior.count = 3.25)

par(mfrow=c(1,2))
multidensity(as.list(as.data.frame(logCPM.MDS_normal)), xlab = "log2 CPM", legend = NULL, main = "Normal samples")
multidensity(as.list(as.data.frame(logCPM.MDS_tumor)), xlab = "log2 CPM", legend = NULL, main = "Tumor samples")

```

Analysing the density graphs from fig. 4, we cannot establish differences between the tumoral/normal logCPM distribution.

### Gene expression distribution

Ending the sample distribution analysis, we can observe the distribution of the log2CPM by gene. We will erase the genes presenting a logCPM below 1, since for these values the CPM the expression level is very low, we cannot establish real conclusions for that genes (spontaneus transcription, non-specific of tissue transcription...).


```{r avgexp gene, fig.align='center', fig.cap='Fig. 5: Histogram presenting gene frequency expression for different logCPM values'}
assays(prac.se.sub)$logCPM <- cpm(prac.dge.unique.filtlib, log=TRUE, prior.count=0.5)

avgexp <- rowMeans(assays(prac.se.sub)$logCPM)

hist(avgexp, xlab="log2 CPM", main="", las=1)

abline(v=1, col="red", lwd=2)

```
As can be observed from fig. 5, there are values for logCPM inferior to 1 (which is considered the minumum standard to consider from). Those values inferior to 1 should not be taken into account when performing the analysis, as those values fall within the Grey Zone (difficulties for differencing data artifacts from the techniques and data obtained from the samples).

### Filtering by gene expression levels

In order to eliminate those values from our, a mask is creating, using as a factor of exclusiong avergae of expression higher than 1. 

```{r gene filt}
mask <- avgexp > 1
dim(prac.se.sub)

prac.se.sub <- prac.se.sub[mask, ]
dim(prac.se.sub)

dim(prac.dge.unique.filtlib )


prac.dge.unique.filtlib <- prac.dge.unique.filtlib[mask, ]
dim(prac.dge.unique.filtlib)

```


<!--- JOAN START --->
## Normalization

First of all, we calculate the normalized factors between sample and we introduce it to prac.se.sub dataset. 

```{r calcNOrm}
prac.dge.unique.filtlib <- calcNormFactors(prac.dge.unique.filtlib)
assays(prac.se.sub)$logCPM <- cpm(prac.dge.unique.filtlib, log=TRUE, prior.count=0.5)
```

From this values, we create an MA plot of each sample to observe the distribution of expression. Unusual samples will be filtered of the dataset later on. 
### MA-plots

<center><h4>Tumor samples MA-plot</h4></center>

```{r maPlotsTumor, fig.align="center", fig.height= 35, fig.width=12, dpi=100, echo=FALSE, fig.cap="Fig. 6: MA-plots of the tumor samples."}

setmp <- prac.se.sub[, prac.se.sub$type == "tumor"]
dgetmp <- prac.dge.unique.filtlib[, prac.dge.unique.filtlib$type == "tumor"]

samples.number <- length(rownames(colData(setmp)))
if (samples.number %% 6 == 0) { 
  nr <- samples.number / 6
} else {nr <- samples.number %/% 6 + 1}
par(mfrow=c(nr, 6), mar=c(4, 5, 3, 1))

for (i in 1:ncol(setmp)) {
  A <- rowMeans(assays(setmp)$logCPM)
  M <- assays(setmp)$logCPM[, i] - A
  samplename <- substr(as.character(setmp$bcr_patient_barcode[i]), 1, 12)
  smoothScatter(A, M, main=samplename, las=1)
  abline(h=0, col="blue", lwd=2)
  lo <- lowess(M ~ A)
  lines(lo$x, lo$y, col="red", lwd=2)
}
```

<center><h4>Normal samples MA-plot</h4></center>

```{r maPlotsNormal, fig.align="center", fig.height= 12.5, fig.width=12, dpi=100, echo=FALSE, fig.cap="Fig. 7: MA-plots of the tumor samples."}

setmp <- prac.se.sub[, prac.se.sub$type == "normal"]
dgetmp <- prac.dge.unique.filtlib[, prac.dge.unique.filtlib$type == "normal"]

samples.number <- length(rownames(colData(setmp)))
if (samples.number %% 6 == 0) { 
  nr <- samples.number / 6
} else {nr <- samples.number %/% 6 + 1}
par(mfrow=c(nr, 6), mar=c(4, 5, 3, 1))

for (i in 1:ncol(setmp)) {
  A <- rowMeans(assays(setmp)$logCPM)
  M <- assays(setmp)$logCPM[, i] - A
  samplename <- substr(as.character(setmp$bcr_patient_barcode[i]), 1, 12)
  smoothScatter(A, M, main=samplename, las=1)
  abline(h=0, col="blue", lwd=2)
  lo <- lowess(M ~ A)
  lines(lo$x, lo$y, col="red", lwd=2)
}
```
As can be percived in fig. 6 and 7, the tendency line representing the mean of normalised values (red line) is reasonably close to the expected tendency line (blue line), which is centered at mean 0.
Despite there are some samples with line tails diverging from the extected values, it is not a big deal, usually.
<!--- JOAN END --->

<!--- DAVID START --->


## Batch Effect

The batch effect occurs when there is any non-biological variable that accounts for the variability between samples. Usually is related to handling, storing and processing the samples. This source of sample variability might obscure the analysis of variation and comparison between samples.

First of all, the batch labels need to be obtaned from the ```se``` object. From it, we obtain the Tissue Source Site (TSS), the center in which the samples where processed, the plate which is sample belonged, and also its portion of analyte and vial.

```{r getting the batch labels, collapse=TRUE}
tss <- substr(colnames(prac.se.sub), 6, 7)
table(tss)
center <- substr(colnames(prac.se.sub), 27, 28)
table(center)
plate <- substr(colnames(prac.se.sub), 22, 25)
table(plate)
portionanalyte <- substr(colnames(prac.se.sub), 18, 20)
table(portionanalyte)
samplevial <- substr(colnames(prac.se.sub), 14, 16)
table(samplevial)
```
<!--- ADRI START --->

By looking at the table between type and tss, we will observe if the data chosen presents batch effect. We have corrected this possibility at the first selections by taking the biggets batches. 

```{r table TSS/TYPE}
table(data.frame(TYPE=prac.se.sub$type, TSS=tss))
```

<!--- ADRI END --->

A good way to see if there is batch effect on the samples is to study the data clustering.

In the following dendogram, samples are being clustered by TSS factor.

```{r hirechical clustering, fig.align='center', fig.width=14, dpi=100, fig.cap='Fig. 8: Hirechical clustering of the prac.se.sub dataset', fig.height=8}
  
par(old.par)
logCPM <- cpm(prac.dge.unique.filtlib, log=TRUE, prior.count=3)
d <- as.dist(1-cor(logCPM, method="spearman"))
sampleClustering <- hclust(d)
batch <- as.integer(factor(tss))
colors <- palette()
palette(c(colors,"darkorange1","darkmagenta","darkolivegreen","red3","seagreen","pink1","yellow3","sienna1"))
sampleDendrogram <- as.dendrogram(sampleClustering, hang=0.1)
names(batch) <- colnames(prac.se.sub)
outcome <- paste(substr(colnames(prac.se.sub), 9, 12), as.character(prac.se.sub$type), sep="-")
names(outcome) <- colnames(prac.se.sub)
sampleDendrogram <- dendrapply(sampleDendrogram,
                               function(x, batch, labels) {
                                 if (is.leaf(x)) {
                                   attr(x, "nodePar") <- list(lab.col=as.vector(batch[attr(x, "label")]))
                                   attr(x, "label") <- as.vector(labels[attr(x, "label")])
                                 }
                                 x
                               }, batch, outcome)

plot(sampleDendrogram, main="Hierarchical clustering of samples")
legend("topright", paste("Batch", sort(unique(batch)), levels(factor(tss))), fill=sort(unique(batch)))

```
When in a dendogram some samles cluster together, it means that they are closer to each other, than the rest of the sample. If we expect no batch effect, the batch labels should be more or less randomly clustered together.
As we can see from the dendogram in fig. 8, some red samples (batch EJ) and blue samples (batch HC) seem to cluster together. This can happen as a result of 2 causes, which are non-mutualy exclusive:

* There are more red and blue samples, thus, it is more likely that they cluster together.
* Red and blue samples are similiar between them, and different between the other samples, hence, clustering together. 

```{r mds, echo=FALSE, fig.align='center', fig.width=14, dpi=100, fig.cap='Fig. 9: MDS of the prac.se.sub dataset', fig.height=8}
par(mfrow=c(1, 1), mar=c(0,0,0,0))
plotMDS(prac.dge.unique.filtlib, labels=outcome, col=batch)
legend("topright", paste("Batch", sort(unique(batch)), levels(factor(tss))),
       fill=sort(unique(batch)), inset=0.05)
```

```{r mds2, echo=FALSE, fig.align='center', fig.width=14, dpi=100, fig.cap='Fig. 10: MDS of the prac.se.sub dataset', fig.height=8}
type <- as.integer(factor(prac.se.sub$type))
plotMDS(prac.dge.unique.filtlib, pch = 19, col=type)
legend("topright", paste("Type", sort(unique(type)), levels(factor(prac.se.sub$type))),
       fill=sort(unique(type)), inset=0.05)
```
MDS-plots help to see also how data cluster together projecting the variability of each sample, in a 2D plot (similar to PCA-plots).
As we  can see in fig. 9 and 10, there is a clustering of normal samples, belonging to batch HC (blue at left-inferior corner). For the sake of keeping the dataset as homogenous as possible, it might be interessting to do not use those samples. Maybe, that mini-cluster of samples is partially responsable of the overall disposition of samples (which differentates poorly between normal and cancer tissues).


<!--- DAVID END --->



<!--- DAVID START ---> 

### Filtering after MDS ands MA plots

As some unexpected results have been found, regarding normal samples, one viable option for the analysis is to remove those samples.

```{r function to prune, echo= FALSE}
GetID <- function(number, se){
  id <- grep(number,rownames(colData(se)))
  return(rownames(colData(prac.se.sub))[id])
}

```

```{r pruning the data set}
number.id <- c("7737","7745","7740","7738","7211","7747","8258") #from MDS plot
pruning.vec <- sapply(number.id, GetID, prac.se.sub)

prac.dge.unique.pruned <- prac.dge.unique.filtlib[,
                   !rownames(prac.dge.unique.filtlib$samples) %in% pruning.vec]
table(prac.dge.unique.pruned$samples$group)
prac.se.pruned <- prac.se.sub[,rownames(prac.dge.unique.pruned$samples)]
```

After pruning, we observe again the distribution of the samples (Fig.11). 

```{r MDS after prunning, echo=FALSE}
tss <- substr(colnames(prac.se.pruned), 6, 7)
batch <- as.integer(factor(tss))
outcome <- paste(substr(colnames(prac.se.pruned), 9, 12), as.character(prac.se.pruned$type), sep="-")
names(outcome) <- colnames(prac.se.pruned)
```

```{r mds3, echo=FALSE, fig.align='center', fig.width=14, dpi=100, fig.cap='Fig. 11: MDS of the prac.se.sub dataset', fig.height=8}
plotMDS(prac.dge.unique.pruned, labels=outcome, col=batch)
legend("topright", paste("Batch", sort(unique(batch)), levels(factor(tss))),
       fill=sort(unique(batch)), inset=0.05)


```

```{r mds4, echo=FALSE, fig.align='center', fig.width=14, dpi=100, fig.cap='MDS of the prac.se.sub dataset', fig.height=8}

type <- as.integer(factor(prac.se.pruned$type))
plotMDS(prac.dge.unique.pruned, pch = 19, col=type)
legend("topright", paste("Type", sort(unique(type)), levels(factor(prac.se.sub$type))),
       fill=sort(unique(type)), inset=0.05)
```


As we can see, the overall clustering of the samples is a bit better than before. Those 7 normal samples were disturbing the clustering by type (normal tissue vs tumor tissue).  


## Differential Gene Expression

After having corrected by those disturbing samples, the differential gene expression can be analysed. The null model created in this case is not adjusted for any attribute (mod0 with intercept 1).


```{r diferential gene expr1}
mod <- model.matrix(~ prac.se.pruned$type, colData(prac.se.pruned))
mod0 <- model.matrix(~ 1, colData(prac.se.pruned))
pv <- f.pvalue(assays(prac.se.pruned)$logCPM, mod, mod0)
sum(p.adjust(pv, method="fdr") < 0.01)
```

```{r pdist, echo=FALSE, out.width="600px", fig.cap="Distribution of raw p-values for an F-test on every gene between tumor and normal samples.", fig.align='center'}


hist(pv, main="", xlab= 'p-value', las=1)

```

The histogram presents a peak in the 0.0-0.1 region, indicating a high presence of DE genes. The distribution of the other values is quite uniform and low in comparison with the 0.1-0.0 values. We apply the SVA modification to observe the changes in the distribution: 


```{r sva 1}
sv <- sva(assays(prac.se.pruned)$logCPM, mod, mod0)
sv$n
```


```{r sva2}
modsv <- cbind(mod, sv$sv)
mod0sv <- cbind(mod0, sv$sv)
pvsv <- f.pvalue(assays(prac.se.pruned)$logCPM, modsv, mod0sv)
sum(p.adjust(pvsv, method="fdr") < 0.01)
```

```{r psvdist, echo=FALSE, out.width="600px", fig.cap="Distribution of raw p-values for an F-test on every gene between tumor and normal samples, adjusting for surrogate variables estimated with SVA.", fig.align='center'}
# pdf("shit2.pdf")
hist(pvsv, main="", xlab= 'p-value', las=1)
# dev.off()
```

The total of DE genes has increased a 27% with respect to the initial number. The distribution of p-values has remained equal, having a decrease in the frequency of each interval (from 400 ocurrences to 300). The uniformity of the distribution is mantained. 

### Checking for batch effect from TSS

As it have not been 100% clear if there is batch effect at the previous step we will use again the sva package to stimate if there exists batch effect by using the TSS as a null model when building the `mod.matrix`.

#### Tss
```{r sva tss}
# introduce the extracted data in the se obj
colData(prac.se.pruned)$tissue_source_site <- tss
table(data.frame(TYPE=prac.se.pruned$type, TSS = tss))
mod <- model.matrix(~ type + as.factor(tissue_source_site), data = colData(prac.se.pruned))
mod0 <- model.matrix(~ as.factor(tissue_source_site), data = colData(prac.se.pruned))
pValues <- f.pvalue(assays(prac.se.pruned)$logCPM, mod, mod0) 
sum(p.adjust(pValues, method = "BH") < 0.01)
```

```{r pvales plot tss, echo=FALSE, fig.align='center', fig.cap= 'p-values of the differential expression analysis using TSS as null model.' }
hist(pValues, xlab= 'p-value')
```

#### Plate
```{r sva plate }
# introduce the extracted data in the se obj
plate <- substr(colnames(prac.se.pruned), 22, 25)
colData(prac.se.pruned)$plate <- plate
mod <- model.matrix(~ type + as.factor(plate), data = colData(prac.se.pruned))
mod0 <- model.matrix(~ as.factor(plate), data = colData(prac.se.pruned))
pValues <- f.pvalue(assays(prac.se.pruned)$logCPM, mod, mod0) 
sum(p.adjust(pValues, method = "BH") < 0.01)
```

```{r pvales plot p, echo=FALSE, fig.align='center', fig.cap= 'p-values of the differential expression analysis using plate as null model.' }
hist(pValues, xlab= 'p-value')
```
#### Samplevial
Trying to test this may be a waste of time, as we know beforehand that sample code makes implicit if the samples come from healty individuals or diseased ones.

In case the test is performed, a high dependance between sample type and sample/vial code will be found, as if they were highly related. In fact, they refer to exactly the same information.

<!--- DAVID END --->


## Conclusion



## Session information

```{r}
sessionInfo()
```
