\documentclass[9pt,twocolumn,twoside]{gsajnl}
% Use the documentclass option 'lineno' to view line numbers

\articletype{inv} % article type
% {inv} Investigation 
% {gs} Genomic Selection
% {goi} Genetics of Immunity 
% {gos} Genetics of Sex 
% {mp} Multiparental Populations


\title{Prostate Adenocarcinoma insights from a Gene Expression approach from RNA-seq data}

\author[$\ast$,$\dagger$]{Adria Auladell-Martin}
\author[$\ast$,$\dagger$]{Joan Marti-Carreras}
\author[$\ast$,$\dagger$]{David Mas-Ponte}
\author[$\ast$,1]{Rober Castelo-Valdueza}

\affil[$\ast$]{M.Sc. in Bioinformatics at Department of Experimental and Health Sciences (CEXS), Universitat Pompeu Fabra}
\affil[$\dagger$]{Authors Contributed Equally to this work}

\keywords{prostate cancer; adenocarcinoma; RNA-Seq; paired data.}

\runningtitle{GENETICS Journal Template on Overleaf} % For use in the footer 

\correspondingauthor{Robert Castelo Valdueza}

\begin{abstract}
The abstract should be written for people who may not read the entire paper, so it must stand on its own. The impression it makes usually determines whether the reader will go on to read the article, so the abstract must be engaging, clear, and concise. In addition, the abstract may be the only part of the article that is indexed in databases, so it must accurately reflect the content of the article. A well-written abstract is the  most effective way to reach intended readers, leading to more robust search, retrieval, and usage of the article. 

\end{abstract}

\setboolean{displaycopyright}{true}

\begin{document}

\maketitle
\thispagestyle{firststyle}
\marginmark
\firstpagefootnote
\correspondingauthoraffiliation{adria.auladell@estudiant.upf.edu, joan.marti02@estudiant.upf.edu, david.mas@estudiant.upf.edu. Corresponding author: Robert Castelo Valdueza, robert.castelo@upf.edu.}
\vspace{-11pt}%

\lettrine[lines=2]{\color{color2}T}{}his \textit{Genetics} journal template is provided to help you write your work in the correct journal format. Instructions for use are provided below. 

\section*{Your Abstract}

In addition to the guidelines provided in the example abstract above, your abstract should:

\begin{itemize}
\item provide a synopsis of the entire article;
\item begin with the broad context of the study, followed by specific background for the study;
\item describe the purpose, methods and procedures, core findings and results, and conclusions of the study;
\item emphasize new or important aspects of the research;
\item engage the broad readership of GENETICS and be understandable to a diverse audience (avoid using jargon);
\item be a single paragraph of less than 250 words;
\item contain the full name of the organism studied;
\item NOT contain citations or abbreviations.
\end{itemize}


\section*{Introduction}
Prostate adenocarcinoma (PRAC), also referred as prostate cancer, is  the second most deadly cancer disease (following lung cancer), and one with the highest prevalences. As the American Cancer Society explains, about 1 man in 7 will be diagnosed with PRAC during his lifespan, and of those diagnosed about 60\% are men over 65 years. Despite it might affect younger men (under 40 years old), it is not frequent. The mean of diagnosis is around 66 years old \citep{prostatestatistics}.

For patients whose cancer has spread, their survival time is usually one to three years. It was estimated that for 2011, 240,890 men would be diagnosed and 33,720 would die from prostate cancer.


Since the first publication of the human genome, multiple international projects have been carried in order to study the genetic bases of the this pathology. One of the most prominent is The Cancer Genome Atlas Project (TCGA) \citep{tgca}. Recently, part of its information was released with part of the original data \cite{Rahman15112015}. TCGA established the following points:

\begin{itemize}
\item 74\% of all tumors being assignable to one of seven molecular classes based on distinct oncogenes drivers:
        fusions involving (1) ERG, (2) ETV1, (3) ETV4, or (4) FLI1 (46\%, 8\%, 4\%, and 1\%, respectively)
        mutations in (5) SPOP or (6) FOXA1; or (7) IDH1. (11\%, 3\%, and 1\%, respectively).
\item 25\% of the prostate cancers had a presumed actionable lesion in the PI3K or MAPK signalling pathways, and DNA repair genes were inactivated in 19\%.
\end{itemize}
   
    
The aim of this study is to provide a second outlook to the released data of the TCGA. For doing so we are to perform a differential expression analysis using paired data from the TCGA.

\section*{Materials and Methods}
The examination of the RNA counts can be performed in an average laptop:
\begin{itemize}
\item CPU Intel Core i5 or superior.
\item 4 Gb of RAM memory or superior.
\item 500 Gb of HDD memory, although it is not a limiting factor.
\end{itemize}

The software used was R version 3.3.0 "Supposedly Educational" \cite{R}. Main packages where download and installed through Bioconductor version 3.3 (BiocInstaller 1.22.2)\cite{bioconductor}.

Detailed information about package versions can be found on the section "Session Info" of the Supplementary Materials.

\subsection*{Data Availability}

The data sets are tables of RNA-seq counts generated by Rahman et al. \cite{Rahman15112015} from the TCGA raw sequence read data using the Rsubread/featureCounts pipeline for all data sets \cite{Rsubread}. They also processed clinical data that forms part of these data sets. PRAC dataset consist initially of 502 tumour samples and 52 normal samples. The phenotypic variable that indicates the tumour or normal status of the sample is called type. 

The RNA-Sequencing and clinical data can be downloaded from Gene Expression Omnibus (accession number GSE62944). Scripts and code that were used to process and analyze the data are available from \url{https://github.com/srp33/TCGA_RNASeq_Clinical}.

Please contact: stephen\_piccolo@byu.edu or andreab@genetics.utah.edu


\subsection*{Statistical Analysis} 
\subsubsection*{Changing dataset}
\subsubsection*{QA, Batch Effect and Normalization}
\subsubsection*{Differential Expression Analysis (DGE)}
\subsubsection*{Functional Analysis}


\section*{Results}
\subsubsection*{Differential Expression Analysis (DGE)}
\subsubsection*{Functional Analysis}

Purament el resultat


\section*{Discussion}

Creure'ns o no els resultats


\section*{Examples of Article Components}
\label{sec:examples}

The sections below show examples of different header levels, which you can use in the primary sections of the manuscript (Results, Discussion, etc.) to organize your content.

\section*{First level section header}

Use this level to group two or more closely related headings in a long article.

\subsection*{Second level section header}

Second level section text.

\subsubsection*{Third level section header:}

Third level section text. These headings may be numbered, but only when the numbers must be cited in the text. 

\section*{Figures and Tables}

Figures and Tables should be labelled and referenced in the standard way using the \verb|\label{}| and \verb|\ref{}| commands.

\subsection*{Sample Figure}

Figure \ref{fig:spectrum} shows an example figure.

\begin{figure}[htbp]
\centering
\includegraphics[width=\linewidth]{example-figure}
\caption{Example figure from \url{10.1534/genetics.114.173807}. Please include your figures in the manuscript for the review process. You can upload figures to Overleaf via the Project menu. Upon acceptance, we'll ask for your figure files to be uploaded in any of the following formats: TIFF (.tiff), JPEG (.jpg), Microsoft PowerPoint (.ppt), EPS (.eps), or Adobe Illustrator (.ai).  Images should be a minimum of 300 dpi in resolution and 500 dpi minimum if line art images.  RGB, CMYK, and Grayscale are all acceptable. Halftones should be high contrast with sharp detail, because some loss of detail and contrast is inevitable in the production process. Figures should be 10-20 cm in width and 1-25 cm in height. Graph axes must be exactly perpendicular and all lines of equal density.
Label multiple figure parts with A, B, etc. in bolded type, and use Arrows and numbers to draw attention to areas you want to highlight. Legends should start with a brief title and should be a self-contained description of the content of the figure that provides enough detail to fully understand the data presented. All conventional symbols used to indicate figure data points are available for typesetting; unconventional symbols should not be used. Italicize all mathematical variables (both in the figure legend and figure) , genotypes, and additional symbols that are normally italicized.  
}%
\label{fig:spectrum}
\end{figure}


\begin{figure}[htbp]
\centering
\includegraphics[width=\linewidth]{example-figure}
\caption{Example movie (the figure file above is used as a placeholder for this example). \textit{GENETICS} supports video and movie files that can be linked from any portion of the article - including the abstract. Acceptable formats include .asf, avi, .wav, and all types of Windows Media files.   
}%
\label{video:spectrum}
\end{figure}


\subsection*{Sample Table}

Table \ref{tab:shape-functions} shows an example table. Avoid shading, color type, line drawings, graphics, or other illustrations within tables. Use tables for data only; present drawings, graphics, and illustrations as separate figures. Histograms should not be used to present data that can be captured easily in text or small tables, as they take up much more space.  

Tables numbers are given in Arabic numerals. Tables should not be numbered 1A, 1B, etc., but if necessary, interior parts of the table can be labeled A, B, etc. for easy reference in the text.  

% latex table generated in R 3.3.0 by xtable 1.8-2 package
% Tue Jun  7 15:58:47 2016
\begin{table*}[htbp]
\centering
\caption{\bf GO enriched terms}
\begin{tableminipage}{\textwidth}
\begin{tabular}{l|c|r|r|r|r|r|p{3cm}|p{3cm}|}
  \hline
 & GOBPID & Pvalue & OddsRatio & ExpCount & Count & Size & Term & Genes \\ 
  \hline
1 & GO:0045078 & 0.00 & 13.78 & 2.70 &   7 &   8 & positive regulation of interferon-gamma biosynthetic process & EBI3, CEBPG, ZFPM1, IL27, IL12B, IL21, CD276 \\ 
  2 & GO:2001044 & 0.00 & 13.78 & 2.70 &   7 &   8 & regulation of integrin-mediated signaling pathway & CTNNA1, EMP2, LIMS2, PRKD1, PHACTR4, ITGB1BP1, CD63 \\ 
  3 & GO:0021670 & 0.01 & 6.89 & 3.04 &   7 &   9 & lateral ventricle development & DBI, DPCD, TSKU, AQP1, PAX5, UCHL5, NUMBL \\ 
  4 & GO:2000008 & 0.01 & 5.25 & 3.71 &   8 &  11 & regulation of protein localization to cell surface & EGF, RAB11FIP5, GPM6B, RANGRF, NRG1, SYNJ2BP, CAV3, RAB11B \\ 
  5 & GO:0030213 & 0.02 & 4.59 & 3.37 &   7 &  10 & hyaluronan biosynthetic process & CLTC, AP2A1, EGF, HAS1, HAS2, HAS3, IL1B \\ 
   \hline
\end{tabular}
 \label{tab:GOenrichment}
\end{tableminipage}
\end{table*}


\begin{table*}[htbp]
\centering
\caption{\bf Students and their grades}
\begin{tableminipage}{\textwidth}
\begin{tabularx}{\textwidth}{XXXX}
\hline
Student & Grade\footnote{This is an example of a footnote in a table. Lowercase, superscript italic letters (a, b, c, etc.) are used by default. You can also use *, **, and *** to indicate conventional levels of statistical significance, explained below the table.} & Rank & Notes \\
\hline
Alice & 82\% & 1 & Performed very well.\\
Bob & 65\% & 3 & Not up to his usual standard.\\
Charlie & 73\% & 2 & A good attempt.\\
\hline
\end{tabularx}
  \label{tab:shape-functions}
\end{tableminipage}
\end{table*}

\bibliography{prac-bibliography}

\end{document}