\documentclass[9pt,twocolumn,twoside]{gsajnl}
% Use the documentclass option 'lineno' to view line numbers

\articletype{inv} % article type
% {inv} Investigation
% {gs} Genomic Selection
% {goi} Genetics of Immunity
% {gos} Genetics of Sex
% {mp} Multiparental Populations


\title{Prostate Adenocarcinoma insights from a Gene Expression approach from RNA-seq data}

\author[$\ast$,$\dagger$]{Adria Auladell-Martin}
\author[$\ast$,$\dagger$]{Joan Marti-Carreras}
\author[$\ast$,$\dagger$]{David Mas-Ponte}
\author[$\ast$,1]{Robert Castelo-Valdueza}

\affil[$\ast$]{M.Sc. in Bioinformatics at Department of Experimental and Health Sciences (CEXS), Universitat Pompeu Fabra}
\affil[$\dagger$]{Authors Contributed Equally to this work}

\keywords{prostate cancer; adenocarcinoma; RNA-Seq; paired data.}

\runningtitle{GENETICS Journal Template on Overleaf} % For use in the footer

\correspondingauthor{Robert Castelo Valdueza}

\begin{abstract}
The abstract should be written for people who may not read the entire paper, so it must stand on its own. The impression it makes usually determines whether the reader will go on to read the article, so the abstract must be engaging, clear, and concise. In addition, the abstract may be the only part of the article that is indexed in databases, so it must accurately reflect the content of the article. A well-written abstract is the  most effective way to reach intended readers, leading to more robust search, retrieval, and usage of the article.

\end{abstract}

\setboolean{displaycopyright}{true}

\begin{document}

\maketitle
\thispagestyle{firststyle}
\marginmark
\firstpagefootnote
\correspondingauthoraffiliation{adria.auladell@estudiant.upf.edu, joan.marti02@estudiant.upf.edu, david.mas@estudiant.upf.edu. Corresponding author: Robert Castelo Valdueza, robert.castelo@upf.edu.}
\vspace{-11pt}%

\lettrine[lines=2]{\color{color2}T}{}his \textit{Genetics} journal template is provided to help you write your work in the correct journal format. Instructions for use are provided below.

\section*{Your Abstract}

In addition to the guidelines provided in the example abstract above, your abstract should:

\begin{itemize}
\item provide a synopsis of the entire article;
\item begin with the broad context of the study, followed by specific background for the study;
\item describe the purpose, methods and procedures, core findings and results, and conclusions of the study;
\item emphasize new or important aspects of the research;
\item engage the broad readership of GENETICS and be understandable to a diverse audience (avoid using jargon);
\item be a single paragraph of less than 250 words;
\item contain the full name of the organism studied;
\item NOT contain citations or abbreviations.
\end{itemize}


\section*{Introduction}
Prostate adenocarcinoma (PRAC), also referred as prostate cancer, is  the second most deadly cancer disease (following lung cancer), and one with the highest prevalences. As the American Cancer Society explains, about 1 man in 7 will be diagnosed with PRAC during his lifespan, and of those diagnosed about 60\% are men over 65 years. Despite it might affect younger men (under 40 years old), it is not frequent. The mean of diagnosis is around 66 years old \cite{prostatestatistics}.

For patients whose cancer has spread, their survival time is usually one to three years. It was estimated that for 2011, 240,890 men would be diagnosed and 33,720 would die from prostate cancer.


Since the first publication of the human genome, multiple international projects have been carried in order to study the genetic bases of the this pathology. One of the most prominent is The Cancer Genome Atlas Project (TCGA) \cite{tgca}. Recently, part of its information was released with part of the original data \cite{Rahman15112015}. TCGA established the following points:

\begin{itemize}
\item 74\% of all tumors being assignable to one of seven molecular classes based on distinct oncogenes drivers:
        fusions involving (1) ERG, (2) ETV1, (3) ETV4, or (4) FLI1 (46\%, 8\%, 4\%, and 1\%, respectively)
        mutations in (5) SPOP or (6) FOXA1; or (7) IDH1. (11\%, 3\%, and 1\%, respectively).
\item 25\% of the prostate cancers had a presumed actionable lesion in the PI3K or MAPK signalling pathways, and DNA repair genes were inactivated in 19\%.
\end{itemize}


The aim of this study is to provide a second outlook to the released data of the TCGA. For doing so we are to perform a differential expression analysis using paired data from the TCGA.

\section*{Materials and Methods}
The examination of the RNA counts can be performed in an average laptop. As a limiting factor for analysing the data, it is recommended to have at least 4 Gb of RAM and CPU Intel Core i5 or superior. HDD memory is not a limiting factor for this analysis.

The software used was R version 3.3.0 "Supposedly Educational" \cite{R}. Main packages where download and installed through Bioconductor version 3.3 (BiocInstaller 1.22.2)\cite{bioconductor}.

Detailed information about package versions can be found on the section "Session Info" of the Supplementary Materials.

In the following sections we describe briefly our analyses. However, a reproducible version of our analysis with all the code elements is also available in the supplementary material.

\subsection*{Data Availability}

The data sets are tables of RNA-seq counts generated by Rahman et al. \cite{Rahman15112015} from the TCGA raw sequence read data using the Rsubread/featureCounts pipeline for all data sets \cite{Rsubread}. They also processed clinical data that forms part of these data sets. PRAC dataset consist initially of 502 tumour samples and 52 normal samples. The phenotypic variable that indicates the tumour or normal status of the sample is called type.

The data was used for this analysis is a variation of the above mentioned datasets, curated by PhD. Robert Castelo-Valdueza \url{http://functionalgenomics.upf.edu/courses/IEO/projects/datasets/sePRAD.rds}.

\subsection*{Experimental Design}
From the initial dataset, there are 52 normal samples and 502 tumour samples. Searching for coincidences in the patients barcode, it was found out that the dataset had mixed paired and unpaired data. It was decided that the paired design was a better choice, as it takes better into account the differences between subjects.

After applying the corresponding logical mask, the dataset was composed by 50 normal samples and 50 tumour samples.

\subsection*{Quality Assessment and Normalization}
DGE functions were used for normalization, which belongs to edgeR package \cite{Robinson01012010}. Normalization factors were calculated and counts per million reads (logCPM) computed.

LogCPM distributions area compared between normal samples and tumour samples. There are no significant differences. Transcripts with logCPM < 1 were deprecated.

Despite there was a peak of high library size in the library size distribution, it was decided to not to take it out. The rest of the distribution was considered to be uniform.

When plotting the log ratio - mean average plots (MA-plots) for all samples, it was noticed that the normalization procedure succeeded. Any deviation was considered as non-significance.

\subsection*{Batch Effect Identification}
Batch effect was evaluated by means of the TCGA barcode information. The code specified center of analysis, tissue source site and plate of the NGS machine. The center was the University of North Carolina for all samples and the tissue source site/plate variables were equilibrated in the tumour/normal comparison (see SM).

To evaluate possible anomalous samples, hierarchical clustering (HC) from \cite{pheatmap} package and multidimensional scaling (MDS) from \cite{limma} package were performed. HC, clustered by Spearman correlation, presented mainly a good clustering, with some of the paired samples clustering together due to the similarities within individual (see SM). MDS plot presented an anomalous clustering of normal samples from the same batch. These samples were filtered out, remaining 42 tumour and 42 normal samples.

\subsection*{Differential Expression Analysis}

A design matrix was built with sample type (tumor/normal). LogCPM measurements were corrected with the  voom (mean-variance relationship of the log-counts) function present in limma package \cite{voom}. As a covariates,  patient barcode and surrogate variables inferred by SVA analysis were included. Surrogate variables are covariates constructed directly from high-dimensional data that can be used in subsequent analyses to adjust for unknown, unmodeled, or latent sources of noise analysis \cite{GSVA}.

With the design matrix and voom logCPM values, Multiple linear regression (MLR) was performed with \cite{limma} package. From MLR analysis, t statistics, moderated F-statistics and  log-odds ratios were caculated with empirical Bayes statistics from \cite{limma} package. P-values were adjusted with FDR measurements and cutoff was established at genes presenting 1\% of FDR. To evaluate results p-values and moderated t-statistics were plotted and DE genes were visualized in volcano plots and  MA plots  from \cite{limma} package (see SM). Top DE hits were visualized with Gviz package. \textcolor{red}{Creieu que cal aixo final?}


\subsection*{Functional Enrichment}


The resulting DE genes from the previous analyses have been tested for functional enrichment using Gene Ontology gene sets. A hyper-geometrical test from the GOstats package \cite{GOstats} was used. As parameters, a p-value cutoff of 0.05 have been set in order to filter only the most significant results. Then, the results have been pruned again for sets with at least 8 genes to avoid extreme results with infinite OddsRatio.

The same analysis have been also performed in the DE over-expressed and under-expressed gene lists to clarify if the GO biological process is indeed enhanced or decreased in tumor cells. The aim of this stratified analysis is to improve the biological content of our results.


\subsection*{Gene Set expression analysis}
In order to analyse differential expression in gene sets and not uniquely at the gene level 2 different approaches have been used, a Gene Set Enrichment Analysis (GSEA) and a Gene Set Variation Analysis (GSVA).

For both GSEA and GSVA the gene sets from the GSVAdata package \cite{GSVAdata} have been selected. After that, the original gene sets were reduced only to curated gene sets  (C2 type) . In this case the KEGG  \cite{kanehisa2016kegg}, Reactome \cite{fabregat2016reactome} and   Biocarta \cite{nishimura2001biocarta} databases were selected. Gene Sets related with Prostate Cancer have been also included from the same data package.


The package GSEABase \cite{GSEABase} was used to define Gene Sets and match gene identifiers to our Entrez Id codes. The classic GSEA consist in the generation of an incidence matrix that let us build a Enrichment Score by that is dependent on how many genes from the gene set are ranked as most DE genes.

The GSVA package \cite{GSVA} was used to define the set expression values. Then, limma  \cite{limma} and SVA packages were used to perform the linear regression and to improve the model respectively \cite{leek2007capturing,svamanual} . Finally, the 30 top DE pathways have been clustered by sample in order to obtain a clear visualization of the ones over and under-expressed in tumor and normal samples.


\section*{Results}
\subsubsection*{Differential Expression Analysis}

After computing the multiple linear model test, and the eBayes test \cite{limma} 	it was found out that:

\begin{itemize}
\item 2168 genes are under-expressed in tumour samples compared to normal samples.
\item 7569 genes have no significant difference in their expression between sample types.
\item 2127 genes are over-expressed in tumour samples compared to normal samples.
\end{itemize}

As there are many variables that can affect the expression data, and not all of them might be taken into account in the analysis or even taken into account by the design of the original project, it was decided that the best option was to use surrogate variable analysis \cite{leek2007capturing}, which is implemented in the SVA package \cite{SVA}.


After defining the null model (bcr patient code variable, which corrects by patients variability) and computing the SV analysis, the volcano plots were performed to graphically represent the distribution of genes by its B (signification) and its log fold changes (intensity).

Genes out of the area formed by minimum levels of signification (depends on FDR) and minimum intensity (expression is doubled).


\begin{figure}[htbp]
\centering
\includegraphics[width=\linewidth]{Volcano.eps}
\caption{Volcano plot of the studies genes. Crossing lines represent the threshold by which genes have been minimally accepted. Most scoring genes are highlighted.
}
\label{fig:Volcano plot}
\end{figure}

Despite that there are lots of genes that fall out of the rejection zones, in fact they conform a gradient of significant gene, only the topmost 5 will be taken into account for the rest of the article, as it is more likely that their values are worth. These 5 genes were then searched using the UCSC browser, using the RefSeq track, as it is more up-to-date.
\begin{itemize}
\item chr1:225 codifies for \href{http://www.genecards.org/cgi-bin/carddisp.pl?gene=EPHA8}{EPHA8} which is an ephrin receptor, a neuronal factor.
\item chr2:101 codifies for \href{http://www.genecards.org/cgi-bin/carddisp.pl?gene=SNORD89}{SNORD89} which is a small nucleolar RNA.
\item chr3:385 codifies for \href{http://www.genecards.org/cgi-bin/carddisp.pl?gene=SCN5A}{SCN5A} which is a sodium channel voltage gated V Type.
\item chr11:11 codifies for \href{http://www.ensembl.org/Homo_sapiens/Gene/Summary?g=ENSG00000255292;r=11:112086903-112193805}{AP002884.2} which is an oxyreductase mitocondrial protein (does not appear on RefSeq).
\item chrX:492 codifies \href{http://www.genecards.org/cgi-bin/carddisp.pl?gene=CACNA1F}{CACNA1F} which is a calcium channel voltage dependant L type.
\end{itemize}

EPHA8 is a prognosis marker for cancer (malignant marker). It appears over-expressed in several tumour types, between them prostate cancer \cite{proteinatlas}.

SNORD89 is related with several cancer-related pathways:  anaplastic large cell lymphomas, lymphoma (NHL) cell lines, metastasis and death cell in colon cancer, prostate cancer, ovarian adenocarcinomas, etc \cite{tcng}.

SCN5A is a voltage-gated sodium channel (VGSC), a group that is known to be related with metastatic potential of breast cancer, prostate cancer and lung cancer between others \cite{Nelson2015}.

AP002884.2 seems to be an unmapped transcript. A more accurated display of the region, reveal that AP002884.2 compromises 2 genes SDHD-006 and BCO2-009. SDHD-006 is the 6th subunit of the succinate dehydrogenase, which is known to play a role as oncometabolite \cite{oncometabolites} and BCO2-009 is related with the synthesis of vitamin A through oxidation of carotenoids  and beta-carotene and have and strong relation with all kinds of cancer \cite{proteinatlas}.

We suggest that in the case of AP002884.2 there has been an error in the mapping phase of the reads. We have considered to take into account both proteins (SDHD-006 and BCO2-009) has both can be traced to carcinogenic processes.

CACNA1F is related with several cancer-related pathways: colon cancer, breast cancer, melanoma, myeloma, prostate cancer, etc \cite{tcng}.

\subsubsection*{Functional Enrichment}
ADRIA
\subsubsection*{Gene set expression analysis}
DAVID


\paragraph{Under-expression of Endogenous Sterols } The genes in this gene set,


Purament el resultat


\section*{Discussion}

Creure'ns o no els resultats

\section*{Figures and Tables}

Figures and Tables should be labelled and referenced in the standard way using the \verb|\label{}| and \verb|\ref{}| commands.

\begin{figure}[htbp]
\centering
\includegraphics[width=\linewidth]{Clustering.eps}
\caption{Initial Cluster of Final data set clustered by gene expression levels. Although some samples are not clustered ideally with their correspondent type, 2 groups can be distinguished clearly. The top-left group contains most of the tumour samples and the bottom-right contains most of the Normal samples.
}%
\label{fig:Clustering}
\end{figure}

\begin{figure}[htbp]
\centering
\includegraphics[width=\linewidth]{ClusteringGSVA.eps}
\caption{Heatmap from the Clustering of the DE gene sets obtained from GSVA and samples.
}
\label{fig:ClusteringGSVA}
\end{figure}




\subsection*{Sample Table}

Table \ref{tab:shape-functions} shows an example table. Avoid shading, color type, line drawings, graphics, or other illustrations within tables. Use tables for data only; present drawings, graphics, and illustrations as separate figures. Histograms should not be used to present data that can be captured easily in text or small tables, as they take up much more space.

Tables numbers are given in Arabic numerals. Tables should not be numbered 1A, 1B, etc., but if necessary, interior parts of the table can be labeled A, B, etc. for easy reference in the text.

% latex table generated in R 3.3.0 by xtable 1.8-2 package
% Tue Jun 14 13:44:38 2016
\begin{table*}[htbp]
\centering
\caption{\bf GO enriched terms}
\begin{tableminipage}{\textwidth}
 \begin{tabular}{|l|c|r|r|r|r|r|p{3cm}|p{3cm}|}
  \hline
 & GOBPID & Pvalue & OddsRatio & ExpCount & Count & Size & Term & Genes \\
  \hline
1 & GO:0032891 & 0.01 & Inf & 4.30 &   8 &   8 & negative regulation of organic acid transport & HRH3, ABAT, AGTR2, AKT2, ACSL4, IL1B, TRH, CYP4F2 \\
  2 & GO:0033189 & 0.01 & 9.47 & 6.45 &  11 &  12 & response to vitamin A & DNMT3A, DNMT3B, GATA4, ARG1, LTC4S, RARA, RXRA, TSHB, TYMS, CAT, ALDH1A2 \\
  3 & GO:0051016 & 0.02 & 7.75 & 5.38 &   9 &  10 & barbed-end actin filament capping & TWF2, ADD1, EPS8, FAM21C, TWF1, VIL1, CAPG, CAPZB, CAPZA3 \\
  4 & GO:0010954 & 0.03 & 6.88 & 4.84 &   8 &   9 & positive regulation of protein processing & ADAM8, F12, HPN, ASTL, PHB, PLGRKT, CNTN2, CLEC3B \\
  5 & GO:0022401 & 0.03 & 6.88 & 4.84 &   8 &   9 & negative adaptation of signaling pathway & ADM, ADRB2, ADRBK1, DNM1, DNM2, DRD3, HTR2B, CAMK2B \\
   \hline
 \end{tabular}
 \label{tab:GOenrichment}
\end{tableminipage}
\end{table*}

\bibliography{prac-bibliography}

\end{document}
